{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "dca877f3",
   "metadata": {},
   "source": [
    "#### Environment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f12606d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7b8591d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda list\n",
    "#list all the included packages."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "0606f5b3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# conda environments:\n",
      "#\n",
      "base                  *  /home/stud03/miniconda3\n",
      "rna_seq_env              /home/stud03/miniconda3/envs/rna_seq_env\n",
      "\n"
     ]
    }
   ],
   "source": [
    "conda env list  # list of environment we have \n",
    "#base env is the default environment\n",
    "# * we are currently working in this environment \n",
    "# it is not necessary to work in another env because I am goona work only in RNASeq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "048f7d8e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# source activate rna_seq_env\n",
    "#if i want to change the env #look at cheat cheet how to create, activate deactivae envs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb223f54",
   "metadata": {},
   "source": [
    "### Installation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "70ae2448",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cd ~\n",
    "cd workspace/rnaseq/tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "dec0abb7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# MultiQC\n",
    "pip3 install multiqc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "649af682",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-02-01 09:37:16--  https://github.com/seqan/flexbar/releases/download/v3.4.0/flexbar-3.4.0-linux.tar.gz\n",
      "Resolving github.com (github.com)... 140.82.121.4\n",
      "Connecting to github.com (github.com)|140.82.121.4|:443... connected.\n",
      "HTTP request sent, awaiting response... 302 Found\n",
      "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/42592882/f3c4c474-77e8-11e8-8bbc-e0dc5cbafac3?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20220201%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20220201T093716Z&X-Amz-Expires=300&X-Amz-Signature=fa65f08daa8de44fb277483fd092d65e402f6182b29f1d478950148c1f43641f&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=42592882&response-content-disposition=attachment%3B%20filename%3Dflexbar-3.4.0-linux.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
      "--2022-02-01 09:37:16--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/42592882/f3c4c474-77e8-11e8-8bbc-e0dc5cbafac3?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20220201%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20220201T093716Z&X-Amz-Expires=300&X-Amz-Signature=fa65f08daa8de44fb277483fd092d65e402f6182b29f1d478950148c1f43641f&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=42592882&response-content-disposition=attachment%3B%20filename%3Dflexbar-3.4.0-linux.tar.gz&response-content-type=application%2Foctet-stream\n",
      "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.111.133, ...\n",
      "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.108.133|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1009921 (986K) [application/octet-stream]\n",
      "Saving to: â€˜flexbar-3.4.0-linux.tar.gzâ€™\n",
      "\n",
      "flexbar-3.4.0-linux 100%[===================>] 986.25K  --.-KB/s    in 0.04s   \n",
      "\n",
      "2022-02-01 09:37:16 (26.5 MB/s) - â€˜flexbar-3.4.0-linux.tar.gzâ€™ saved [1009921/1009921]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "wget https://github.com/seqan/flexbar/releases/download/v3.4.0/flexbar-3.4.0-linux.tar.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "740478ef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "flexbar-3.4.0-linux/\n",
      "flexbar-3.4.0-linux/flexbar\n",
      "flexbar-3.4.0-linux/libtbb.so.2\n",
      "flexbar-3.4.0-linux/LICENSE.md\n",
      "flexbar-3.4.0-linux/README.md\n"
     ]
    }
   ],
   "source": [
    "tar -xzvf flexbar-3.4.0-linux.tar.gz\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c56a181e",
   "metadata": {},
   "source": [
    "#### Quality control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "6450d943",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ~"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f850b17c",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd workspace/rnaseq/RawData"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "75fdb492",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Command 'parallel' not found, but can be installed with:\n",
      "\n",
      "apt install moreutils  # version 0.63-1, or\n",
      "apt install parallel   # version 20161222-1.1\n",
      "\n",
      "Ask your administrator to install one of them.\n",
      "\n"
     ]
    },
    {
     "ename": "",
     "evalue": "127",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "find . -name \"*.fastq.gz\" | parallel fastqc -o {//}/ {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a1477072",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1;31mE: \u001b[0mCould not open lock file /var/lib/dpkg/lock-frontend - open (13: Permission denied)\u001b[0m\n",
      "\u001b[1;31mE: \u001b[0mUnable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), are you root?\u001b[0m\n"
     ]
    },
    {
     "ename": "",
     "evalue": "100",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "apt install parallel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "262d1966",
   "metadata": {},
   "outputs": [],
   "source": [
    "fastqc *.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "49b161d1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28\n"
     ]
    }
   ],
   "source": [
    "ls *.gz | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "bb2bb351",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28\n"
     ]
    }
   ],
   "source": [
    "ls *.html | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "ef7e9448",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/stud03/workspace/rnaseq/RawData\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "d86d47e7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  \u001b[34m/\u001b[0m\u001b[32m/\u001b[0m\u001b[31m/\u001b[0m \u001b]8;id=922937;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ ðŸ” \u001b[2m| v1.11\u001b[0m\n",
      "\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /home/stud03/workspace/rnaseq/RawData\n",
      "\u001b[2K\u001b[34m|\u001b[0m         \u001b[34msearching\u001b[0m | \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m84/84\u001b[0m  \u001b[0m.html\u001b[0m.html\u001b[0m.html\u001b[0m\n",
      "\u001b[?25h\u001b[34m|            fastqc\u001b[0m | Found 28 reports\n",
      "\u001b[34m|           multiqc\u001b[0m | Compressing plot data\n",
      "\u001b[34m|           multiqc\u001b[0m | Report      : multiqc_report.html\n",
      "\u001b[34m|           multiqc\u001b[0m | Data        : multiqc_data\n",
      "\u001b[34m|           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "multiqc . #Run MultiQC on your fastqc reports to generate a single summary report across all samples/replicates."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "40028de3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/stud03\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c7f57309",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd workspace/rnaseq/rawData"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "efbe45b9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mRNA_NHEM_76_ACAGTG_L007_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_NHEM_76_ACAGTG_L007_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_NHEM_77_GCCAAT_L007_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_NHEM_77_GCCAAT_L007_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_Sbcl2_64_TAGCTT_L007_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_Sbcl2_64_TAGCTT_L007_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_Sbcl2_70_TGACCA_L008_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_Sbcl2_70_TGACCA_L008_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_68_CGATGT_L008_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_68_CGATGT_L008_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_74_ACTTGA_L007_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_74_ACTTGA_L007_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1366_66_CTTGTA_L007_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1366_66_CTTGTA_L007_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1366_72_GCCAAT_L008_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1366_72_GCCAAT_L008_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM3211_65_GGCTAC_L007_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM3211_65_GGCTAC_L007_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM3211_71_ACAGTG_L008_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM3211_71_ACAGTG_L008_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM793_67_ATCACG_L008_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM793_67_ATCACG_L008_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM793_73_CAGATC_L008_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM793_73_CAGATC_L008_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM9_69_TTAGGC_L008_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM9_69_TTAGGC_L008_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM9_75_GATCAG_L007_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM9_75_GATCAG_L007_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;34mmultiqc_data\u001b[0m\n",
      "\u001b[01;34mqc\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09cca9ed",
   "metadata": {},
   "source": [
    "#### Adapter Triming"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "28717f7f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-02-01 09:14:28--  http://genomedata.org/rnaseq-tutorial/illumina_multiplex.fa\n",
      "Resolving genomedata.org (genomedata.org)... 54.71.55.4\n",
      "Connecting to genomedata.org (genomedata.org)|54.71.55.4|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 161\n",
      "Saving to: â€˜illumina_multiplex.faâ€™\n",
      "\n",
      "illumina_multiplex. 100%[===================>]     161  --.-KB/s    in 0s      \n",
      "\n",
      "2022-02-01 09:14:28 (15.7 MB/s) - â€˜illumina_multiplex.faâ€™ saved [161/161]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "wget http://genomedata.org/rnaseq-tutorial/illumina_multiplex.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a2bff9c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "RARY_PATH=$/student_tools/flexbar-3.4.0-linux:"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e2a31c9",
   "metadata": {},
   "source": [
    "/home/stud03/workspace/rnaseq/rawDataTrim/illumina_multiplex.fa\n",
    "/home/stud03/workspace/rnaseq/RawData/RNA_NHEM_76_ACAGTG_L007_R1_001.fastq.gz\n",
    "/home/stud03/workspace/rnaseq/RawData/RNA_NHEM_76_ACAGTG_L007_R2_001.fastq.gz\n",
    "/home/stud03/workspace/rnaseq/rawData/TrimRNA_NHEM_76_ACAGTG_L007\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9d303ad3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5ce73198",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/stud03/miniconda3\n",
      "\n",
      "  added / updated specs:\n",
      "    - trimmomatic\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    dbus-1.13.18               |       hb2f20db_0         504 KB\n",
      "    expat-2.4.1                |       h2531618_2         168 KB\n",
      "    freetype-2.11.0            |       h70c0345_0         618 KB\n",
      "    glib-2.69.1                |       h4ff587b_1         1.7 MB\n",
      "    libpng-1.6.37              |       hbc83047_0         278 KB\n",
      "    libxcb-1.14                |       h7b6447c_0         505 KB\n",
      "    openjdk-11.0.13            |       h87a67e3_0       341.0 MB\n",
      "    pcre-8.45                  |       h295c915_0         207 KB\n",
      "    trimmomatic-0.39           |       hdfd78af_2         144 KB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:       345.1 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  dbus               pkgs/main/linux-64::dbus-1.13.18-hb2f20db_0\n",
      "  expat              pkgs/main/linux-64::expat-2.4.1-h2531618_2\n",
      "  freetype           pkgs/main/linux-64::freetype-2.11.0-h70c0345_0\n",
      "  glib               pkgs/main/linux-64::glib-2.69.1-h4ff587b_1\n",
      "  libpng             pkgs/main/linux-64::libpng-1.6.37-hbc83047_0\n",
      "  libxcb             pkgs/main/linux-64::libxcb-1.14-h7b6447c_0\n",
      "  openjdk            pkgs/main/linux-64::openjdk-11.0.13-h87a67e3_0\n",
      "  pcre               pkgs/main/linux-64::pcre-8.45-h295c915_0\n",
      "  trimmomatic        bioconda/noarch::trimmomatic-0.39-hdfd78af_2\n",
      "\n",
      "\n",
      "Proceed ([y]/n)? \n",
      "CondaSystemExit: \n",
      "Operation aborted.  Exiting.\n",
      "\n",
      "(base) "
     ]
    }
   ],
   "source": [
    "conda install trimmomatic\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b9ca264e",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd /home/stud03/workspace/rnaseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4be2ba6d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34mRNA_REFS_DIR\u001b[0m  \u001b[01;34mrawData\u001b[0m  \u001b[01;34mtools\u001b[0m  \u001b[01;34mtrimData\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "0251bce3",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd trimData"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "a4fd0103",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RNA_NHEM_76_trimmed.log                  \u001b[0m\u001b[01;31mRNA_WM1366_72_trimmed_1.fastq.gz\u001b[0m\n",
      "\u001b[01;31mRNA_NHEM_76_trimmed_1.fastq.gz\u001b[0m           \u001b[01;31mRNA_WM1366_72_trimmed_2.fastq.gz\u001b[0m\n",
      "\u001b[01;31mRNA_NHEM_76_trimmed_2.fastq.gz\u001b[0m           RNA_WM3211_65_trimmed.log\n",
      "RNA_NHEM_77_trimmed.log                  \u001b[01;31mRNA_WM3211_65_trimmed_1.fastq.gz\u001b[0m\n",
      "\u001b[01;31mRNA_NHEM_77_trimmed_1.fastq.gz\u001b[0m           \u001b[01;31mRNA_WM3211_65_trimmed_2.fastq.gz\u001b[0m\n",
      "\u001b[01;31mRNA_NHEM_77_trimmed_2.fastq.gz\u001b[0m           RNA_WM3211_71_trimmed.log\n",
      "RNA_Sbcl2_64_trimmed.log                 \u001b[01;31mRNA_WM3211_71_trimmed_1.fastq.gz\u001b[0m\n",
      "\u001b[01;31mRNA_Sbcl2_64_trimmed_1.fastq.gz\u001b[0m          \u001b[01;31mRNA_WM3211_71_trimmed_2.fastq.gz\u001b[0m\n",
      "\u001b[01;31mRNA_Sbcl2_64_trimmed_2.fastq.gz\u001b[0m          RNA_WM793_67_trimmed.log\n",
      "RNA_Sbcl2_70_trimmed.log                 \u001b[01;31mRNA_WM793_67_trimmed_1.fastq.gz\u001b[0m\n",
      "\u001b[01;31mRNA_Sbcl2_70_trimmed_1.fastq.gz\u001b[0m          \u001b[01;31mRNA_WM793_67_trimmed_2.fastq.gz\u001b[0m\n",
      "\u001b[01;31mRNA_Sbcl2_70_trimmed_2.fastq.gz\u001b[0m          RNA_WM793_73_trimmed.log\n",
      "RNA_WM1158_68_CGATGT_L008.log            \u001b[01;31mRNA_WM793_73_trimmed_1.fastq.gz\u001b[0m\n",
      "\u001b[01;31mRNA_WM1158_68_CGATGT_L008_1.fastq.gz\u001b[0m     \u001b[01;31mRNA_WM793_73_trimmed_2.fastq.gz\u001b[0m\n",
      "RNA_WM1158_68_CGATGT_L008_1_fastqc.html  RNA_WM9_69_trimmed.log\n",
      "\u001b[01;31mRNA_WM1158_68_CGATGT_L008_1_fastqc.zip\u001b[0m   \u001b[01;31mRNA_WM9_69_trimmed_1.fastq.gz\u001b[0m\n",
      "\u001b[01;31mRNA_WM1158_68_CGATGT_L008_2.fastq.gz\u001b[0m     \u001b[01;31mRNA_WM9_69_trimmed_2.fastq.gz\u001b[0m\n",
      "RNA_WM1158_68_CGATGT_L008_2_fastqc.html  RNA_WM9_75_trimmed.log\n",
      "\u001b[01;31mRNA_WM1158_68_CGATGT_L008_2_fastqc.zip\u001b[0m   \u001b[01;31mRNA_WM9_75_trimmed_1.fastq.gz\u001b[0m\n",
      "RNA_WM1158_74_Trimmed.log                \u001b[01;31mRNA_WM9_75_trimmed_2.fastq.gz\u001b[0m\n",
      "\u001b[01;31mRNA_WM1158_74_Trimmed_1.fastq.gz\u001b[0m         TruSeq2-PE.fa\n",
      "\u001b[01;31mRNA_WM1158_74_Trimmed_2.fastq.gz\u001b[0m         TruSeq3-PE.fa\n",
      "RNA_WM1366_66_trimmed.log                \u001b[01;34madapters\u001b[0m\n",
      "\u001b[01;31mRNA_WM1366_66_trimmed_1.fastq.gz\u001b[0m         illumina_mine.fa\n",
      "\u001b[01;31mRNA_WM1366_66_trimmed_2.fastq.gz\u001b[0m         illumina_multiplex.fa\n",
      "RNA_WM1366_72_trimmed.log\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "17d6fa1f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">PrefixPE/1\n",
      "TACACTCTTTCCCTACACGACGCTCTTCCGATCT\n",
      ">PrefixPE/2\n",
      "GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT"
     ]
    }
   ],
   "source": [
    "cat TruSeq3-PE.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c2fc3b0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/trimData/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_NHEM_76_ACAGTG_L007_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/rawData/RNA_NHEM_76_ACAGTG_L007_R2_001.fastq.gz --target ~/workspace/rnaseq/trimData/RNA_NHEM_76_trimmed -qt 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d9785f16",
   "metadata": {},
   "outputs": [],
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/trimData/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_NHEM_77_GCCAAT_L007_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/rawData/RNA_NHEM_77_GCCAAT_L007_R1_001.fastq.gz --target ~/workspace/rnaseq/rawDataTrim/RNA_NHEM_77_trimmed -qt 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2fb961a",
   "metadata": {},
   "outputs": [],
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/trimData/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_Sbcl2_64_TAGCTT_L007_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/rawData/RNA_Sbcl2_64_TAGCTT_L007_R2_001.fastq.gz --target ~/workspace/rnaseq/trimData/RNA_Sbcl2_64_trimmed -qt 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2db10e10",
   "metadata": {},
   "outputs": [],
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/trimData/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_Sbcl2_70_TGACCA_L008_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/rawData/RNA_Sbcl2_70_TGACCA_L008_R2_001.fastq.gz --target ~/workspace/rnaseq/trimData/RNA_Sbcl2_70_trimmed -qt 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1713e30d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "0451d650",
   "metadata": {},
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/NAS_MEB/trimData/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/NAS_MEB/qc/RNA_Sbcl2_64_TAGCTT_L007_R1_001.fastq.gz --reads2 ~/NAS_MEB/qc/RNA_Sbcl2_64_TAGCTT_L007_R2_001.fastq.gz  --target ~/NAS_MEB/qc/RNA_Sbcl2_64_trimmed -qt 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1f22e059",
   "metadata": {},
   "outputs": [],
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/NAS_MEB/trimData/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/NAS_MEB/qc/RNA_Sbcl2_70_TGACCA_L008_R1_001.fastq.gz --reads2 ~/NAS_MEB/qc/RNA_Sbcl2_70_TGACCA_L008_R2_001.fastq.gz --target ~/NAS_MEB/qc/RNA_Sbcl2_70_trimmed -qt 30"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3c4cda43",
   "metadata": {},
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/rawDataTrim/illumina_multiplex.fa --pre-trim-left 13 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/RawData/RNA_WM1158_74_ACTTGA_L007_R1_001.fastq.gz\n",
    "--reads2 ~/workspace/rnaseq/RawData/RNA_WM1158_74_ACTTGA_L007_R2_001.fastq.gz --target ~/workspace/rnaseq/rawDataTrim/TrimRNA_NHEM_76_ACAGTG_L007"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "673c57be",
   "metadata": {},
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/rawDataTrim/TruSeq3-PE.fa --pre-trim-left 8 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_WM1158_68_CGATGT_L008_R1_001.fastq.gz  --reads2 /workspace/rnaseq/rawData/RNA_WM1158_68_CGATGT_L008_R2_001.fastq.gz --target ~/workspace/rnaseq/rawDataTrim/TrimRNA_Sbcl2_70 -qt 30"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f65ac5a6",
   "metadata": {},
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/rawDataTrim/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_WM1158_68_CGATGT_L008_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/rawData/RNA_WM1158_68_CGATGT_L008_R2_001.fastq.gz --target ~/workspace/rnaseq/rawDataTrim/RNA_WM1158_68_CGATGT_L008 -qt 30"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "da1551c9",
   "metadata": {},
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/rawDataTrim/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_WM1158_68_CGATGT_L008_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/rawData/RNA_WM1158_68_CGATGT_L008_R2_001.fastq.gz --target ~/workspace/rnaseq/rawDataTrim/RNA_WM1158_68_CGATGT_L008 -qt 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c54b4e2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/rawDataTrim/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_WM1158_74_ACTTGA_L007_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/rawData/RNA_WM1158_74_ACTTGA_L007_R2_001.fastq.gz --target ~/workspace/rnaseq/rawDataTrim/RNA_WM1158_74_Trimmed -qt 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "568087a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/rawDataTrim/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_WM1366_66_CTTGTA_L007_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/rawData/RNA_WM1366_66_CTTGTA_L007_R2_001.fastq.gz --target ~/workspace/rnaseq/rawDataTrim/RNA_WM1366_66_trimmed -qt 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "284aea2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/rawDataTrim/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_WM1366_72_GCCAAT_L008_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/rawData/RNA_WM1366_72_GCCAAT_L008_R2_001.fastq.gz --target ~/workspace/rnaseq/rawDataTrim/RNA_WM1366_72_trimmed -qt 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "7530911d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/stud03/workspace/rnaseq/rawData\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6667838f",
   "metadata": {},
   "outputs": [],
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/rawDataTrim/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_WM3211_65_GGCTAC_L007_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/rawData/RNA_WM3211_65_GGCTAC_L007_R2_001.fastq.gz --target ~/workspace/rnaseq/rawDataTrim/RNA_WM3211_65_trimmed -qt 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "89e89c3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/rawDataTrim/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_WM3211_71_ACAGTG_L008_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/rawData/RNA_WM3211_71_ACAGTG_L008_R2_001.fastq.gz --target ~/workspace/rnaseq/rawDataTrim/RNA_WM3211_71_trimmed -qt 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "ccaa41f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/rawDataTrim/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_WM793_67_ATCACG_L008_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/rawData/RNA_WM793_67_ATCACG_L008_R2_001.fastq.gz --target ~/workspace/rnaseq/rawDataTrim/RNA_WM793_67_trimmed -qt 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "a808a9db",
   "metadata": {},
   "outputs": [],
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/rawDataTrim/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_WM793_73_CAGATC_L008_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/rawData/RNA_WM793_73_CAGATC_L008_R2_001.fastq.gz --target ~/workspace/rnaseq/rawDataTrim/RNA_WM793_73_trimmed -qt 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cefbf640",
   "metadata": {},
   "outputs": [],
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/rawDataTrim/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_WM9_69_TTAGGC_L008_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/rawData/RNA_WM9_69_TTAGGC_L008_R2_001.fastq.gz --target ~/workspace/rnaseq/rawDataTrim/RNA_WM9_69_trimmed -qt 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "83d15989",
   "metadata": {},
   "outputs": [],
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/rawDataTrim/illumina_mine.fa --pre-trim-left 10 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/rawData/RNA_WM9_75_GATCAG_L007_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/rawData/RNA_WM9_75_GATCAG_L007_R1_001.fastq.gz --target ~/workspace/rnaseq/rawDataTrim/RNA_WM9_75_trimmed -qt 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "caea0be7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RNA_NHEM_76.log                \u001b[0m\u001b[01;31mTrimRNA_NHEM_77_2.fastq.gz\u001b[0m\n",
      "RNA_Sbcl2_70                   TrimRNA_NHEM_77_2_fastqc.html\n",
      "TrimRNA_NHEM_76.log            \u001b[01;31mTrimRNA_NHEM_77_2_fastqc.zip\u001b[0m\n",
      "\u001b[01;31mTrimRNA_NHEM_76_1.fastq.gz\u001b[0m     TrimRNA_Sbcl2_64.log\n",
      "TrimRNA_NHEM_76_1_fastqc.html  \u001b[01;31mTrimRNA_Sbcl2_64_1.fastq.gz\u001b[0m\n",
      "\u001b[01;31mTrimRNA_NHEM_76_1_fastqc.zip\u001b[0m   TrimRNA_Sbcl2_64_1_fastqc.html\n",
      "\u001b[01;31mTrimRNA_NHEM_76_2.fastq.gz\u001b[0m     \u001b[01;31mTrimRNA_Sbcl2_64_1_fastqc.zip\u001b[0m\n",
      "TrimRNA_NHEM_76_2_fastqc.html  \u001b[01;31mTrimRNA_Sbcl2_64_2.fastq.gz\u001b[0m\n",
      "\u001b[01;31mTrimRNA_NHEM_76_2_fastqc.zip\u001b[0m   TrimRNA_Sbcl2_64_2_fastqc.html\n",
      "TrimRNA_NHEM_77.log            \u001b[01;31mTrimRNA_Sbcl2_64_2_fastqc.zip\u001b[0m\n",
      "\u001b[01;31mTrimRNA_NHEM_77_1.fastq.gz\u001b[0m     TruSeq2-PE.fa\n",
      "TrimRNA_NHEM_77_1_fastqc.html  \u001b[01;34madapters\u001b[0m\n",
      "\u001b[01;31mTrimRNA_NHEM_77_1_fastqc.zip\u001b[0m   illumina_multiplex.fa\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1e4f33fa",
   "metadata": {},
   "source": [
    "##### Trying with Trimmomatic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "c16ef9e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters ~/workspace/rnaseq/rawDataTrim/TruSeq2-PE.fa --pre-trim-left 13 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads ~/workspace/rnaseq/RawData/RNA_Sbcl2_64_TAGCTT_L007_R1_001.fastq.gz --reads2 ~/workspace/rnaseq/RawData/RNA_Sbcl2_64_TAGCTT_L007_R2_001.fastq.gz --target ~/workspace/rnaseq/rawDataTrim/TrimRNA_Sbcl2_64"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d15e071b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Usage: \n",
      "       PE [-version] [-threads <threads>] [-phred33|-phred64] [-trimlog <trimLogFile>] [-summary <statsSummaryFile>] [-quiet] [-validatePairs] [-basein <inputBase> | <inputFile1> <inputFile2>] [-baseout <outputBase> | <outputFile1P> <outputFile1U> <outputFile2P> <outputFile2U>] <trimmer1>...\n",
      "   or: \n",
      "       SE [-version] [-threads <threads>] [-phred33|-phred64] [-trimlog <trimLogFile>] [-summary <statsSummaryFile>] [-quiet] <inputFile> <outputFile> <trimmer1>...\n",
      "   or: \n",
      "       -version\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "trimmomatic --help\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "215e8267",
   "metadata": {},
   "source": [
    "trimmomatic PE -threads 5 ~/workspace/rnaseq/RawData/RNA_Sbcl2_70_TGACCA_L008_R1_001.fastq.gz ~/workspace/rnaseq/RawData/RNA_Sbcl2_70_TGACCA_L008_R2_001.fastq.gz  RNA_Sbcl2_70_TGACCA_L008_R1_p.fastq.gz RNA_Sbcl2_70_TGACCA_L008_R1_up.fastq.gz RNA_Sbcl2_70_TGACCA_L008_R2_p.fastq.gz RNA_Sbcl2_70_TGACCA_L008_R2_up.fastq.gz  ILLUMINACLIP:~/workspace/rnaseq/rawDataTrim/adapters/TruSeq2-PE.fa2 LEADING:3 TRAILING:3 MINLEN:36"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56421b13",
   "metadata": {},
   "outputs": [],
   "source": [
    "SLIDINGWINDOW:4:20 MINLEN:25 ILLUMINACLIP:NexteraPE-PE.fa:2:40:15"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "9d42ea65",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "99ea1c65",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ~/workspace/rnaseq/dataTrim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c547fd78",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34mGenomeIndices\u001b[0m                      Untitled2.ipynb\n",
      "Master_Work.ipynb                  \u001b[01;34mflexbar-3.4.0-linux\u001b[0m\n",
      "Miniconda3-latest-Linux-x86_64.sh  \u001b[01;31mflexbar-3.4.0-linux.tar.gz\u001b[0m\n",
      "\u001b[01;34mNAS_MEB\u001b[0m                            \u001b[01;34mminiconda3\u001b[0m\n",
      "Untitled.ipynb                     w\n",
      "Untitled1.ipynb                    \u001b[01;34mworkspace\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "88bb35e7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 8 /home/stud03/workspace/rnaseq/RawData/RNA_Sbcl2_70_TGACCA_L008_R1_001.fastq.gz /home/stud03/workspace/rnaseq/RawData/RNA_Sbcl2_70_TGACCA_L008_R2_001.fastq.gz /home/stud03/workspace/rnaseq/dataTrim/RNA_Sbcl2_70_TGACCA_L008_R1_001.paired.fastq.gz /home/stud03/workspace/rnaseq/dataTrim/RNA_Sbcl2_70_TGACCA_L008_R1_001.unpaired.fastq.gz /home/stud03/workspace/rnaseq/dataTrim/RNA_Sbcl2_70_TGACCA_L008_R2_001.paired.fastq.gz /home/stud03/workspace/rnaseq/dataTrim/RNA_Sbcl2_70_TGACCA_L008_R2_001.unpaired.fastq.gz ILLUMINACLIP:~/workspace/rnaseq/rawDataTrim/TruSeq2-PE.fa/:2:30:10 LEADING:5 TRAILING:5 SLIDINGWINDOW:5:10 MINLEN:50\n",
      "java.io.FileNotFoundException: /home/stud03/~/workspace/rnaseq/rawDataTrim/TruSeq2-PE.fa (No such file or directory)\n",
      "\tat java.base/java.io.FileInputStream.open0(Native Method)\n",
      "\tat java.base/java.io.FileInputStream.open(FileInputStream.java:219)\n",
      "\tat java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)\n",
      "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
      "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
      "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
      "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
      "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
      "\tat org.usadellab.trimmomatic.TrimmomaticPE.run(TrimmomaticPE.java:552)\n",
      "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:80)\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 20812501 Both Surviving: 20343358 (97.75%) Forward Only Surviving: 360632 (1.73%) Reverse Only Surviving: 93674 (0.45%) Dropped: 14837 (0.07%)\n",
      "TrimmomaticPE: Completed successfully\n"
     ]
    }
   ],
   "source": [
    "trimmomatic PE -threads 8 ~/workspace/rnaseq/RawData/RNA_Sbcl2_70_TGACCA_L008_R1_001.fastq.gz ~/workspace/rnaseq/RawData/RNA_Sbcl2_70_TGACCA_L008_R2_001.fastq.gz /home/stud03/workspace/rnaseq/dataTrim/RNA_Sbcl2_70_TGACCA_L008_R1_001.paired.fastq.gz /home/stud03/workspace/rnaseq/dataTrim/RNA_Sbcl2_70_TGACCA_L008_R1_001.unpaired.fastq.gz /home/stud03/workspace/rnaseq/dataTrim/RNA_Sbcl2_70_TGACCA_L008_R2_001.paired.fastq.gz /home/stud03/workspace/rnaseq/dataTrim/RNA_Sbcl2_70_TGACCA_L008_R2_001.unpaired.fastq.gz  ILLUMINACLIP:~/workspace/rnaseq/rawDataTrim/TruSeq2-PE.fa/:2:30:10 LEADING:5 TRAILING:5 SLIDINGWINDOW:5:10 MINLEN:50"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ab8f9e6e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34mGenomeIndices\u001b[0m                      Untitled2.ipynb\n",
      "Master_Work.ipynb                  \u001b[01;34mflexbar-3.4.0-linux\u001b[0m\n",
      "Miniconda3-latest-Linux-x86_64.sh  \u001b[01;31mflexbar-3.4.0-linux.tar.gz\u001b[0m\n",
      "\u001b[01;34mNAS_MEB\u001b[0m                            \u001b[01;34mminiconda3\u001b[0m\n",
      "Untitled.ipynb                     \u001b[01;34mworkspace\u001b[0m\n",
      "Untitled1.ipynb\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e4423779",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "c88f64bc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/stud03/miniconda3\n",
      "\n",
      "  added / updated specs:\n",
      "    - star\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    star-2.7.10a               |       h9ee0642_0         4.9 MB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         4.9 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  star               bioconda/linux-64::star-2.7.10a-h9ee0642_0\n",
      "\n",
      "\n",
      "Proceed ([y]/n)? \n",
      "CondaSystemExit: \n",
      "Operation aborted.  Exiting.\n",
      "\n",
      "(base) "
     ]
    }
   ],
   "source": [
    "conda install STAR"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "876c0697",
   "metadata": {},
   "source": [
    "#### Alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "13ed13dc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/stud03/NAS_MEB/workspace/rnaseq\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ee7c577d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mRNA_NHEM_76_ACAGTG_L007_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_NHEM_76_ACAGTG_L007_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_NHEM_77_GCCAAT_L007_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_NHEM_77_GCCAAT_L007_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_Sbcl2_64_TAGCTT_L007_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_Sbcl2_64_TAGCTT_L007_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_Sbcl2_70_TGACCA_L008_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_Sbcl2_70_TGACCA_L008_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_68_CGATGT_L008_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_68_CGATGT_L008_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_74_ACTTGA_L007_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_74_ACTTGA_L007_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1366_66_CTTGTA_L007_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1366_66_CTTGTA_L007_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1366_72_GCCAAT_L008_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM1366_72_GCCAAT_L008_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM3211_65_GGCTAC_L007_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM3211_65_GGCTAC_L007_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM3211_71_ACAGTG_L008_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM3211_71_ACAGTG_L008_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM793_67_ATCACG_L008_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM793_67_ATCACG_L008_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM793_73_CAGATC_L008_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM793_73_CAGATC_L008_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM9_69_TTAGGC_L008_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM9_69_TTAGGC_L008_R2_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM9_75_GATCAG_L007_R1_001.fastq.gz\u001b[0m\n",
      "\u001b[01;32mRNA_WM9_75_GATCAG_L007_R2_001.fastq.gz\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "8cf8e50c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2021, December 22). GNU Parallel 20211222 ('StÃ¸jberg').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.5797028\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "parallel -j 1 echo {1} ::: RNA_NHEM_76_ACAGTG_L007_R1_001 RNA_NHEM_76_ACAGTG_L007_R2_001 NA_NHEM_77_GCCAAT_L007_R1_001 RNA_NHEM_77_GCCAAT_L007_R2_001 RNA_Sbcl2_64_TAGCTT_L007_R1_001 RNA_Sbcl2_64_TAGCTT_L007_R2_001 RNA_Sbcl2_70_TGACCA_L008_R1_001 RNA_Sbcl2_70_TGACCA_L008_R2_001 RNA_WM1158_68_CGATGT_L008_R1_001.fastq.gz RNA_WM1158_68_CGATGT_L008_R2_001 RNA_WM1158_74_ACTTGA_L007_R1_001 RNA_WM1158_74_ACTTGA_L007_R2_001 RNA_WM1366_66_CTTGTA_L007_R1_001 RNA_WM1366_66_CTTGTA_L007_R2_001 RNA_WM1366_72_GCCAAT_L008_R1_001 RNA_WM1366_72_GCCAAT_L008_R2_001 RNA_WM3211_65_GGCTAC_L007_R1_001 RNA_WM3211_65_GGCTAC_L007_R2_001 RNA_WM3211_71_ACAGTG_L008_R1_001 RNA_WM3211_71_ACAGTG_L008_R2_001 RNA_WM793_67_ATCACG_L008_R1_001 RNA_WM793_67_ATCACG_L008_R2_001 RNA_WM793_73_CAGATC_L008_R1_001 RNA_WM793_73_CAGATC_L008_R2_001 RNA_WM9_69_TTAGGC_L008_R1_001 RNA_WM9_69_TTAGGC_L008_R2_001 RNA_WM9_75_GATCAG_L007_R1_001 RNA_WM9_75_GATCAG_L007_R2_001 ::: > /home/stud03/NAS_MEB/workspace/rnaseq/SampleNames_fastq.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2d343ed3",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd NAS_MEB/workspace/rnaseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5255cbac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34mRNASeq_Melanoma_Samples\u001b[0m  \u001b[01;32mSampleNames_fastq.txt\u001b[0m  \u001b[01;34mgeneIndex\u001b[0m  \u001b[01;34mtools\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ae862831",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RNA_NHEM_76_ACAGTG_L007_R1_001\n",
      "RNA_NHEM_76_ACAGTG_L007_R2_001\n",
      "NA_NHEM_77_GCCAAT_L007_R1_001\n",
      "RNA_NHEM_77_GCCAAT_L007_R2_001\n",
      "RNA_Sbcl2_64_TAGCTT_L007_R1_001\n",
      "RNA_Sbcl2_64_TAGCTT_L007_R2_001\n",
      "RNA_Sbcl2_70_TGACCA_L008_R1_001\n",
      "RNA_Sbcl2_70_TGACCA_L008_R2_001\n",
      "RNA_WM1158_68_CGATGT_L008_R1_001\n",
      "RNA_WM1158_68_CGATGT_L008_R2_001\n",
      "RNA_WM1158_74_ACTTGA_L007_R1_001\n",
      "RNA_WM1158_74_ACTTGA_L007_R2_001\n",
      "RNA_WM1366_66_CTTGTA_L007_R1_001\n",
      "RNA_WM1366_66_CTTGTA_L007_R2_001\n",
      "RNA_WM1366_72_GCCAAT_L008_R1_001\n",
      "RNA_WM1366_72_GCCAAT_L008_R2_001\n",
      "RNA_WM3211_65_GGCTAC_L007_R1_001\n",
      "RNA_WM3211_65_GGCTAC_L007_R2_001\n",
      "RNA_WM3211_71_ACAGTG_L008_R1_001\n",
      "RNA_WM3211_71_ACAGTG_L008_R2_001\n",
      "RNA_WM793_67_ATCACG_L008_R1_001\n",
      "RNA_WM793_67_ATCACG_L008_R2_001\n",
      "RNA_WM793_73_CAGATC_L008_R1_001\n",
      "RNA_WM793_73_CAGATC_L008_R2_001\n",
      "RNA_WM9_69_TTAGGC_L008_R1_001\n",
      "RNA_WM9_69_TTAGGC_L008_R2_001\n",
      "RNA_WM9_75_GATCAG_L007_R1_001\n",
      "RNA_WM9_75_GATCAG_L007_R2_001\n"
     ]
    }
   ],
   "source": [
    "cat SampleNames_fastq.txt\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3cd2027d",
   "metadata": {},
   "source": [
    "### Download Ref genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4dcfd459",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd refGenome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "2aac72a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-04-04 13:43:58--  https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_genomic.fna.gz\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 165.112.9.228, 165.112.9.229, 2607:f220:41f:250::228, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|165.112.9.228|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 964894122 (920M) [application/x-gzip]\n",
      "Saving to: â€˜GRCh38_latest_genomic.fna.gzâ€™\n",
      "\n",
      "GRCh38_latest_genom 100%[===================>] 920.19M  10.8MB/s    in 83s     \n",
      "\n",
      "2022-04-04 13:45:21 (11.1 MB/s) - â€˜GRCh38_latest_genomic.fna.gzâ€™ saved [964894122/964894122]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "wget https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_genomic.fna.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "9fbc21a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "gunzip GRCh38_latest_genomic.fna.gz #extract fasta nucleic acid file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "baf1dba3",
   "metadata": {},
   "outputs": [],
   "source": [
    "cp GRCh38_latest_genomic.fna GRCh38.fa #rename fasta file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "0d5d8a0d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">NC_000001.11 Homo sapiens chromosome 1, GRCh38.p13 Primary Assembly\n",
      "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n"
     ]
    }
   ],
   "source": [
    "head -10 GRCh38.fa "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "5172c320",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-04-04 13:57:17--  https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_genomic.gff.gz\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.10, 165.112.9.230, 2607:f220:41e:250::13, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.10|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 51127410 (49M) [application/x-gzip]\n",
      "Saving to: â€˜GRCh38_latest_genomic.gff.gzâ€™\n",
      "\n",
      "GRCh38_latest_genom 100%[===================>]  48.76M  10.0MB/s    in 5.5s    \n",
      "\n",
      "2022-04-04 13:57:23 (8.92 MB/s) - â€˜GRCh38_latest_genomic.gff.gzâ€™ saved [51127410/51127410]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "wget https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_genomic.gff.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "16933873",
   "metadata": {},
   "outputs": [],
   "source": [
    "gunzip *gff.gz #Extract gff annotation file"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0fb6048a",
   "metadata": {},
   "source": [
    "#### Ensembl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "baf1532f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/stud03/NAS_MEB/workspace/rnaseq/refGenome\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c0a5bb62",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34mRNASeq_Melanoma_Samples\u001b[0m  \u001b[01;34mgeneIndex\u001b[0m  \u001b[01;34mrefGenome_ensembl\u001b[0m\n",
      "\u001b[01;32mSampleNames_fastq.txt\u001b[0m    \u001b[01;34mrefGenome\u001b[0m  \u001b[01;34mtools\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "6fe5c867",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!genome-build GRCh38.p13\n",
      "#!genome-version GRCh38\n",
      "#!genome-date 2013-12\n",
      "#!genome-build-accession GCA_000001405.28\n",
      "#!genebuild-last-updated 2021-08\n",
      "1\tensembl_havana\tgene\t1211340\t1214153\t.\t-\t.\tgene_id \"ENSG00000186827\"; gene_version \"11\"; gene_name \"TNFRSF4\"; gene_source \"ensembl_havana\"; gene_biotype \"protein_coding\";\n",
      "1\tensembl_havana\ttranscript\t1211340\t1214153\t.\t-\t.\tgene_id \"ENSG00000186827\"; gene_version \"11\"; transcript_id \"ENST00000379236\"; transcript_version \"4\"; gene_name \"TNFRSF4\"; gene_source \"ensembl_havana\"; gene_biotype \"protein_coding\"; transcript_name \"TNFRSF4-201\"; transcript_source \"ensembl_havana\"; transcript_biotype \"protein_coding\"; tag \"CCDS\"; ccds_id \"CCDS11\"; tag \"basic\"; transcript_support_level \"1 (assigned to previous version 3)\";\n",
      "1\tensembl_havana\texon\t1213983\t1214153\t.\t-\t.\tgene_id \"ENSG00000186827\"; gene_version \"11\"; transcript_id \"ENST00000379236\"; transcript_version \"4\"; exon_number \"1\"; gene_name \"TNFRSF4\"; gene_source \"ensembl_havana\"; gene_biotype \"protein_coding\"; transcript_name \"TNFRSF4-201\"; transcript_source \"ensembl_havana\"; transcript_biotype \"protein_coding\"; tag \"CCDS\"; ccds_id \"CCDS11\"; exon_id \"ENSE00001832731\"; exon_version \"2\"; tag \"basic\"; transcript_support_level \"1 (assigned to previous version 3)\";\n",
      "1\tensembl_havana\tCDS\t1213983\t1214127\t.\t-\t0\tgene_id \"ENSG00000186827\"; gene_version \"11\"; transcript_id \"ENST00000379236\"; transcript_version \"4\"; exon_number \"1\"; gene_name \"TNFRSF4\"; gene_source \"ensembl_havana\"; gene_biotype \"protein_coding\"; transcript_name \"TNFRSF4-201\"; transcript_source \"ensembl_havana\"; transcript_biotype \"protein_coding\"; tag \"CCDS\"; ccds_id \"CCDS11\"; protein_id \"ENSP00000368538\"; protein_version \"3\"; tag \"basic\"; transcript_support_level \"1 (assigned to previous version 3)\";\n",
      "1\tensembl_havana\tstart_codon\t1214125\t1214127\t.\t-\t0\tgene_id \"ENSG00000186827\"; gene_version \"11\"; transcript_id \"ENST00000379236\"; transcript_version \"4\"; exon_number \"1\"; gene_name \"TNFRSF4\"; gene_source \"ensembl_havana\"; gene_biotype \"protein_coding\"; transcript_name \"TNFRSF4-201\"; transcript_source \"ensembl_havana\"; transcript_biotype \"protein_coding\"; tag \"CCDS\"; ccds_id \"CCDS11\"; tag \"basic\"; transcript_support_level \"1 (assigned to previous version 3)\";\n"
     ]
    }
   ],
   "source": [
    "head  Homo_sapiens.GRCh38.105.gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f1ce16dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd refGenome_ensembl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "29b101a0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 59G\n",
      "-rwxrwxr-x 1 stud03 stud2 1.3G Apr  4 14:13 \u001b[0m\u001b[01;32mHomo_sapiens.GRCh38.105.gtf\u001b[0m\n",
      "-rwxrwxr-x 1 stud03 stud2  57G Apr  4 14:53 \u001b[01;32mHomo_sapiens.GRCh38.dna.alt.fa\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls -lh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "9ae0f5e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-04-04 14:11:36--  http://ftp.ensembl.org/pub/release-105/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.alt.fa.gz\n",
      "Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.197.76\n",
      "Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.197.76|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 226443084 (216M) [application/x-gzip]\n",
      "Saving to: â€˜Homo_sapiens.GRCh38.dna.alt.fa.gzâ€™\n",
      "\n",
      "Homo_sapiens.GRCh38 100%[===================>] 215.95M  2.63MB/s    in 83s     \n",
      "\n",
      "2022-04-04 14:12:59 (2.61 MB/s) - â€˜Homo_sapiens.GRCh38.dna.alt.fa.gzâ€™ saved [226443084/226443084]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "wget http://ftp.ensembl.org/pub/release-105/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.alt.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "78e7d66b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-04-04 14:13:33--  http://ftp.ensembl.org/pub/release-105/gtf/homo_sapiens/Homo_sapiens.GRCh38.105.gtf.gz\n",
      "Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.197.76\n",
      "Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.197.76|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 50829040 (48M) [application/x-gzip]\n",
      "Saving to: â€˜Homo_sapiens.GRCh38.105.gtf.gzâ€™\n",
      "\n",
      "Homo_sapiens.GRCh38 100%[===================>]  48.47M  2.55MB/s    in 19s     \n",
      "\n",
      "2022-04-04 14:13:52 (2.60 MB/s) - â€˜Homo_sapiens.GRCh38.105.gtf.gzâ€™ saved [50829040/50829040]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "wget http://ftp.ensembl.org/pub/release-105/gtf/homo_sapiens/Homo_sapiens.GRCh38.105.gtf.gz # download anno file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "0fcfb084",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/stud03/NAS_MEB/workspace/rnaseq/refGenome_ensembl\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "41da7661",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mHomo_sapiens.GRCh38.105.gtf.gz\u001b[0m  \u001b[01;32mHomo_sapiens.GRCh38.dna.alt.fa.gz\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#extract both files\n",
    "gunzip *.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5ca6e7db",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd NAS_MEB/workspace/rnaseq/refGenome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "8fc3e75b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34mRNASeq_Melanoma_Samples\u001b[0m  \u001b[01;34mgeneIndex\u001b[0m    \u001b[01;34mrefGenome\u001b[0m          \u001b[01;34mtools\u001b[0m\n",
      "\u001b[01;32mSampleNames_fastq.txt\u001b[0m    \u001b[01;34mgeneIndices\u001b[0m  \u001b[01;34mrefGenome_ensembl\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1cf98370",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tSTAR --runMode genomeGenerate --runThreadN 14 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --genomeFastaFiles /home/stud03/NAS_MEB/workspace/rnaseq/refGenome_ensembl/Homo_sapiens.GRCh38.dna.alt.fa --sjdbGTFfile /home/stud03/NAS_MEB/workspace/rnaseq/refGenome_ensembl/Homo_sapiens.GRCh38.105.gtf --sjdbOverhang 99 --limitGenomeGenerateRAM 24000000000 --genomeSAsparseD 2\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 07 22:59:02 ..... started STAR run\n",
      "!!!!! WARNING: Could not move Log.out file from ./Log.out into /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices/Log.out. Will keep ./Log.out\n",
      "\n",
      "Apr 07 22:59:03 ... starting to generate Genome files\n",
      "\n"
     ]
    }
   ],
   "source": [
    "STAR --runMode genomeGenerate --runThreadN 14 --genomeDir ~/NAS_MEB/workspace/rnaseq/geneIndices --genomeFastaFiles ~/NAS_MEB/workspace/rnaseq/refGenome_ensembl/Homo_sapiens.GRCh38.dna.alt.fa --sjdbGTFfile ~/NAS_MEB/workspace/rnaseq/refGenome_ensembl/Homo_sapiens.GRCh38.105.gtf --sjdbOverhang 99 --limitGenomeGenerateRAM 24000000000 --genomeSAsparseD 2\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "583d34bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd NAS_MEB/workspace/rnaseq/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1054bda2",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir genome_ebi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "434076a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-04-07 23:03:23--  ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_29/GRCh38.primary_assembly.genome.fa.gz\n",
      "           => â€˜GRCh38.primary_assembly.genome.fa.gzâ€™\n",
      "Resolving ftp.ebi.ac.uk (ftp.ebi.ac.uk)... 193.62.193.138\n",
      "Connecting to ftp.ebi.ac.uk (ftp.ebi.ac.uk)|193.62.193.138|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /pub/databases/gencode/Gencode_human/release_29 ... done.\n",
      "==> SIZE GRCh38.primary_assembly.genome.fa.gz ... 844691642\n",
      "==> PASV ... done.    ==> RETR GRCh38.primary_assembly.genome.fa.gz ... done.\n",
      "Length: 844691642 (806M) (unauthoritative)\n",
      "\n",
      "GRCh38.primary_asse 100%[===================>] 805.56M  11.6MB/s    in 69s     \n",
      "\n",
      "2022-04-07 23:04:33 (11.7 MB/s) - â€˜GRCh38.primary_assembly.genome.fa.gzâ€™ saved [844691642]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_29/GRCh38.primary_assembly.genome.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5c7d3b7d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-04-07 23:04:56--  ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_29/gencode.v29.annotation.gtf.gz\n",
      "           => â€˜gencode.v29.annotation.gtf.gzâ€™\n",
      "Resolving ftp.ebi.ac.uk (ftp.ebi.ac.uk)... 193.62.197.74\n",
      "Connecting to ftp.ebi.ac.uk (ftp.ebi.ac.uk)|193.62.197.74|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /pub/databases/gencode/Gencode_human/release_29 ... done.\n",
      "==> SIZE gencode.v29.annotation.gtf.gz ... 39387922\n",
      "==> PASV ... done.    ==> RETR gencode.v29.annotation.gtf.gz ... done.\n",
      "Length: 39387922 (38M) (unauthoritative)\n",
      "\n",
      "gencode.v29.annotat 100%[===================>]  37.56M  9.56MB/s    in 4.1s    \n",
      "\n",
      "2022-04-07 23:05:01 (9.25 MB/s) - â€˜gencode.v29.annotation.gtf.gzâ€™ saved [39387922]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_29/gencode.v29.annotation.gtf.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d75ade25",
   "metadata": {},
   "outputs": [],
   "source": [
    "gunzip GRCh38.primary_assembly.genome.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "063dbe89",
   "metadata": {},
   "outputs": [],
   "source": [
    "gunzip gencode.v29.annotation.gtf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "5f90aac2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mGRCh38.primary_assembly.genome.fa\u001b[0m  \u001b[01;32mgencode.v29.annotation.gtf\u001b[0m\n",
      "/home/stud03/NAS_MEB/workspace/rnaseq/genome_ebi\n"
     ]
    }
   ],
   "source": [
    "ls\n",
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ffda4292",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tSTAR --runMode genomeGenerate --runThreadN 24 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --genomeFastaFiles /home/stud03/NAS_MEB/workspace/rnaseq/genome_ebi/GRCh38.primary_assembly.genome.fa --sjdbGTFfile /home/stud03/NAS_MEB/workspace/rnaseq/genome_ebi/gencode.v29.annotation.gtf --sjdbOverhang 99\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 07 23:20:52 ..... started STAR run\n",
      "!!!!! WARNING: Could not move Log.out file from ./Log.out into /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices/Log.out. Will keep ./Log.out\n",
      "\n",
      "Apr 07 23:20:52 ... starting to generate Genome files\n",
      "Apr 07 23:32:39 ..... processing annotations GTF\n",
      "Apr 07 23:37:28 ... starting to sort Suffix Array. This may take a long time...\n",
      "Apr 07 23:37:43 ... sorting Suffix Array chunks and saving them to disk...\n",
      "Apr 07 23:58:44 ... loading chunks from disk, packing SA...\n",
      "Apr 08 00:11:31 ... finished generating suffix array\n",
      "Apr 08 00:11:31 ... generating Suffix Array index\n",
      "Apr 08 00:15:23 ... completed Suffix Array index\n",
      "Apr 08 00:15:27 ..... inserting junctions into the genome indices\n",
      "Apr 08 00:18:35 ... writing Genome to disk ...\n",
      "Apr 08 00:19:23 ... writing Suffix Array to disk ...\n",
      "Apr 08 00:25:40 ... writing SAindex to disk\n",
      "Apr 08 00:26:04 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "STAR --runMode genomeGenerate --runThreadN 24 --genomeDir ~/NAS_MEB/workspace/rnaseq/geneIndices --genomeFastaFiles ~/NAS_MEB/workspace/rnaseq/genome_ebi/GRCh38.primary_assembly.genome.fa --sjdbGTFfile ~/NAS_MEB/workspace/rnaseq/genome_ebi/gencode.v29.annotation.gtf --sjdbOverhang 99"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "088cb257",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd NAS_MEB/workspace/rnaseq/geneIndices"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9508de9f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mGenome\u001b[0m         \u001b[01;32mchrNameLength.txt\u001b[0m  \u001b[01;32mgenomeParameters.txt\u001b[0m\n",
      "\u001b[01;32mSA\u001b[0m             \u001b[01;32mchrStart.txt\u001b[0m       \u001b[01;32msjdbInfo.txt\u001b[0m\n",
      "\u001b[01;32mSAindex\u001b[0m        \u001b[01;32mexonGeTrInfo.tab\u001b[0m   \u001b[01;32msjdbList.fromGTF.out.tab\u001b[0m\n",
      "\u001b[01;32mchrLength.txt\u001b[0m  \u001b[01;32mexonInfo.tab\u001b[0m       \u001b[01;32msjdbList.out.tab\u001b[0m\n",
      "\u001b[01;32mchrName.txt\u001b[0m    \u001b[01;32mgeneInfo.tab\u001b[0m       \u001b[01;32mtranscriptInfo.tab\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b221f5b3",
   "metadata": {},
   "source": [
    "#### Mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ffa1b6e2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2021, December 22). GNU Parallel 20211222 ('StÃ¸jberg').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.5797028\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n",
      "\tSTAR --runThreadN 15 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_NHEM_76_ACAGTG_L007_R1_001.fastq /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_NHEM_76_ACAGTG_L007_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 08 22:24:02 ..... started STAR run\n",
      "Apr 08 22:24:02 ..... loading genome\n",
      "Apr 08 22:33:16 ..... started mapping\n",
      "Apr 08 22:46:01 ..... finished mapping\n",
      "Apr 08 22:46:03 ..... started sorting BAM\n",
      "Apr 08 22:46:48 ..... started wiggle output\n",
      "Apr 08 22:49:40 ..... finished successfully\n",
      "\tSTAR --runThreadN 15 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_NHEM_77_GCCAAT_L007_R1_001.fastq /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_NHEM_77_GCCAAT_L007_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 08 22:49:42 ..... started STAR run\n",
      "Apr 08 22:49:42 ..... loading genome\n",
      "Apr 08 22:56:58 ..... started mapping\n",
      "Apr 08 23:16:25 ..... finished mapping\n",
      "Apr 08 23:16:27 ..... started sorting BAM\n",
      "Apr 08 23:17:31 ..... started wiggle output\n",
      "Apr 08 23:20:47 ..... finished successfully\n",
      "\tSTAR --runThreadN 15 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_Sbcl2_64_TAGCTT_L007_R1_001.fastq /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_Sbcl2_64_TAGCTT_L007_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 08 23:20:49 ..... started STAR run\n",
      "Apr 08 23:20:49 ..... loading genome\n",
      "Apr 08 23:28:06 ..... started mapping\n",
      "Apr 08 23:42:07 ..... finished mapping\n",
      "Apr 08 23:42:09 ..... started sorting BAM\n",
      "Apr 08 23:42:56 ..... started wiggle output\n",
      "Apr 08 23:45:42 ..... finished successfully\n",
      "\tSTAR --runThreadN 15 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_Sbcl2_70_TGACCA_L008_R1_001.fastq /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_Sbcl2_70_TGACCA_L008_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 08 23:45:44 ..... started STAR run\n",
      "Apr 08 23:45:44 ..... loading genome\n",
      "Apr 08 23:52:56 ..... started mapping\n",
      "Apr 09 00:11:02 ..... finished mapping\n",
      "Apr 09 00:11:03 ..... started sorting BAM\n",
      "Apr 09 00:11:51 ..... started wiggle output\n",
      "Apr 09 00:14:42 ..... finished successfully\n",
      "\tSTAR --runThreadN 15 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM1158_68_CGATGT_L008_R1_001.fastq /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM1158_68_CGATGT_L008_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 09 00:14:44 ..... started STAR run\n",
      "Apr 09 00:14:44 ..... loading genome\n",
      "Apr 09 00:21:55 ..... started mapping\n",
      "Apr 09 00:39:56 ..... finished mapping\n",
      "Apr 09 00:39:58 ..... started sorting BAM\n",
      "Apr 09 00:41:01 ..... started wiggle output\n",
      "Apr 09 00:44:18 ..... finished successfully\n",
      "\tSTAR --runThreadN 15 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM1158_74_ACTTGA_L007_R1_001.fastq /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM1158_74_ACTTGA_L007_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 09 00:44:20 ..... started STAR run\n",
      "Apr 09 00:44:20 ..... loading genome\n",
      "Apr 09 00:51:31 ..... started mapping\n",
      "Apr 09 01:00:26 ..... finished mapping\n",
      "Apr 09 01:00:28 ..... started sorting BAM\n",
      "Apr 09 01:00:56 ..... started wiggle output\n",
      "Apr 09 01:03:18 ..... finished successfully\n",
      "\tSTAR --runThreadN 15 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM1366_66_CTTGTA_L007_R1_001.fastq /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM1366_66_CTTGTA_L007_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 09 01:03:20 ..... started STAR run\n",
      "Apr 09 01:03:20 ..... loading genome\n",
      "Apr 09 01:10:30 ..... started mapping\n",
      "Apr 09 01:18:27 ..... finished mapping\n",
      "Apr 09 01:18:29 ..... started sorting BAM\n",
      "Apr 09 01:18:56 ..... started wiggle output\n",
      "Apr 09 01:21:18 ..... finished successfully\n",
      "parallel: Error: Output is incomplete.\n",
      "parallel: Error: Cannot append to buffer file in /tmp.\n",
      "parallel: Error: Is the disk full?\n",
      "parallel: Error: Change $TMPDIR with --tmpdir or use --compress.\n"
     ]
    },
    {
     "ename": "",
     "evalue": "255",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cat ~/NAS_MEB/workspace/rnaseq/SampleNames_fastq.txt | parallel -j 1 \"mkdir ~/workspace/rnaseq/Mapped/{} ; cd ~/workspace/rnaseq/Mapped/{} ; STAR --runThreadN 15 --genomeDir  ~/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn ~/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/{}_R1_001.fastq ~/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/{}_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a1c7e36d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2021, December 22). GNU Parallel 20211222 ('StÃ¸jberg').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.5797028\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n",
      "\tSTAR --runThreadN 15 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM1366_72_GCCAAT_L008_R1_001.fastq /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM1366_72_GCCAAT_L008_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 09 02:51:23 ..... started STAR run\n",
      "Apr 09 02:51:23 ..... loading genome\n",
      "Apr 09 02:59:06 ..... started mapping\n",
      "Apr 09 03:22:46 ..... finished mapping\n",
      "Apr 09 03:24:45 ..... started sorting BAM\n",
      "Apr 09 03:30:57 ..... started wiggle output\n",
      "Apr 09 03:43:09 ..... finished successfully\n",
      "mkdir: cannot create directory â€˜/home/stud03/NAS_MEB/workspace/rnaseq/Mapped/RNA_WM1366_72_GCCAAT_L008â€™: File exists\n",
      "\tSTAR --runThreadN 15 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM3211_65_GGCTAC_L007_R1_001.fastq /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM3211_65_GGCTAC_L007_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 09 03:43:14 ..... started STAR run\n",
      "Apr 09 03:43:14 ..... loading genome\n",
      "Apr 09 03:50:26 ..... started mapping\n",
      "Apr 09 04:02:13 ..... finished mapping\n",
      "Apr 09 04:03:40 ..... started sorting BAM\n",
      "Apr 09 04:07:03 ..... started wiggle output\n",
      "Apr 09 04:14:54 ..... finished successfully\n",
      "\tSTAR --runThreadN 15 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM3211_71_ACAGTG_L008_R1_001.fastq /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM3211_71_ACAGTG_L008_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 09 04:14:58 ..... started STAR run\n",
      "Apr 09 04:14:58 ..... loading genome\n",
      "Apr 09 04:22:12 ..... started mapping\n",
      "Apr 09 04:37:44 ..... finished mapping\n",
      "Apr 09 04:39:27 ..... started sorting BAM\n",
      "Apr 09 04:44:13 ..... started wiggle output\n",
      "Apr 09 04:53:26 ..... finished successfully\n",
      "\tSTAR --runThreadN 15 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM793_67_ATCACG_L008_R1_001.fastq /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM793_67_ATCACG_L008_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 09 04:53:30 ..... started STAR run\n",
      "Apr 09 04:53:30 ..... loading genome\n",
      "Apr 09 05:00:41 ..... started mapping\n",
      "Apr 09 05:16:11 ..... finished mapping\n",
      "Apr 09 05:17:58 ..... started sorting BAM\n",
      "Apr 09 05:22:19 ..... started wiggle output\n",
      "Apr 09 05:31:35 ..... finished successfully\n",
      "\tSTAR --runThreadN 15 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM793_73_CAGATC_L008_R1_001.fastq /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM793_73_CAGATC_L008_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 09 05:31:39 ..... started STAR run\n",
      "Apr 09 05:31:39 ..... loading genome\n",
      "Apr 09 05:38:53 ..... started mapping\n",
      "Apr 09 05:56:19 ..... finished mapping\n",
      "Apr 09 05:58:08 ..... started sorting BAM\n",
      "Apr 09 06:02:33 ..... started wiggle output\n",
      "Apr 09 06:12:38 ..... finished successfully\n",
      "\tSTAR --runThreadN 15 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM9_69_TTAGGC_L008_R1_001.fastq /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM9_69_TTAGGC_L008_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 09 06:12:41 ..... started STAR run\n",
      "Apr 09 06:12:41 ..... loading genome\n",
      "Apr 09 06:19:54 ..... started mapping\n",
      "Apr 09 06:44:01 ..... finished mapping\n",
      "Apr 09 06:45:47 ..... started sorting BAM\n",
      "Apr 09 06:53:36 ..... started wiggle output\n",
      "Apr 09 07:07:48 ..... finished successfully\n",
      "\tSTAR --runThreadN 15 --genomeDir /home/stud03/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM9_75_GATCAG_L007_R1_001.fastq /home/stud03/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/RNA_WM9_75_GATCAG_L007_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Apr 09 07:07:53 ..... started STAR run\n",
      "Apr 09 07:07:53 ..... loading genome\n",
      "Apr 09 07:15:16 ..... started mapping\n",
      "Apr 09 07:24:05 ..... finished mapping\n",
      "Apr 09 07:25:21 ..... started sorting BAM\n",
      "Apr 09 07:27:00 ..... started wiggle output\n",
      "Apr 09 07:31:58 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "cat ~/NAS_MEB/workspace/rnaseq/SampleNames2_fastq.txt | parallel -j 1 \"mkdir ~/NAS_MEB/workspace/rnaseq/Mapped/{} ; cd ~/NAS_MEB/workspace/rnaseq/Mapped/{} ; STAR --runThreadN 15 --genomeDir  ~/NAS_MEB/workspace/rnaseq/geneIndices --readFilesIn ~/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/{}_R1_001.fastq ~/NAS_MEB/workspace/rnaseq/RNASeq_Melanoma_Samples/{}_R2_001.fastq --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterIntronMotifs RemoveNoncanonicalUnannotated --outReadsUnmapped Fastq --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --alignMatesGapMax 1000000 --alignIntronMax 1000000 --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outWigType bedGraph --outWigStrand Stranded\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "928a8ce6",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd NAS_MEB/workspace/rnaseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "fc764d92",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34mMapped\u001b[0m                   \u001b[01;34mcountTable2\u001b[0m                \u001b[01;34mgeneIndices\u001b[0m\n",
      "\u001b[01;34mMapping\u001b[0m                  \u001b[01;34mcountTable3\u001b[0m                \u001b[01;34mgenome_ebi\u001b[0m\n",
      "\u001b[01;34mRNASeq_Melanoma_Samples\u001b[0m  \u001b[01;34mcountTable4\u001b[0m                \u001b[01;32mhg38.refGene.gtf\u001b[0m\n",
      "\u001b[01;32mSampleNames2_fastq.txt\u001b[0m   \u001b[01;32mfeaturecounts.txt\u001b[0m          \u001b[01;34mrefGenome\u001b[0m\n",
      "\u001b[01;32mSampleNames_fastq.txt\u001b[0m    \u001b[01;32mfeaturecounts.txt.summary\u001b[0m  \u001b[01;34mrefGenome_ensembl\u001b[0m\n",
      "\u001b[01;34mcountTable\u001b[0m               \u001b[01;34mgeneIndex\u001b[0m                  \u001b[01;34mtools\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1301bab3",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir countTable4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4fdf21e2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.0\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m14 BAM files  \u001b[0m \u001b[0m                                  ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mfeaturecounts.txt\u001b[0m \u001b[0m                               ||\n",
      "||                 Summary : \u001b[36mfeaturecounts.txt.summary\u001b[0m \u001b[0m                       ||\n",
      "||              Annotation : \u001b[36mgencode.v29.annotation.gtf (GTF)\u001b[0m \u001b[0m                ||\n",
      "||      Dir for temp files : \u001b[36m/home/stud03/NAS_MEB/workspace/rnaseq\u001b[0m \u001b[0m           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m23\u001b[0m \u001b[0m                                              ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36myes\u001b[0m \u001b[0m                                             ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||          Chimeric reads : \u001b[36mcounted\u001b[0m \u001b[0m                                         ||\n",
      "||        Both ends mapped : \u001b[36mnot required\u001b[0m \u001b[0m                                    ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file gencode.v29.annotation.gtf \u001b[0m... \u001b[0m                       ||\n",
      "||    Features : \u001b[36m1262773\u001b[0m \u001b[0m                                                     ||\n",
      "||    Meta-features : \u001b[36m58721\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m25\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m16383290\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m9920772 (60.6%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m7.95 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m23298019\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m14628815 (62.8%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m11.87 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m17249142\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m9102556 (52.8%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m9.69 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m16135620\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m8933548 (55.4%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m7.51 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m22995282\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m11427146 (49.7%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m15.06 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m10630552\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m5838804 (54.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m5.33 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m9607522\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m5276454 (54.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m5.37 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m21347066\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m11218527 (52.6%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m15.54 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m11252174\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m6467365 (57.5%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m6.10 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m16249540\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m8412118 (51.8%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m11.85 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m14508316\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m9111122 (62.8%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m7.80 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m14171986\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m8528542 (60.2%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m6.73 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m26494372\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m14340018 (54.1%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m17.20 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m5695766\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m3221739 (56.6%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m2.81 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"/home/stud03/NAS_MEB/wo \u001b[0m ||\n",
      "|| rkspace/rnaseq/featurecounts.txt.summary\" \u001b[0m                                 ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#featureCounts -T 23 -p -t exon -g gene_id -a ~/NAS_MEB/workspace/rnaseq/genome_ebi/gencode.v29.annotation.gtf -o ~/NAS_MEB/workspace/rnaseq/countTable4/featurecounts.txt ~/NAS_MEB/workspace/rnaseq/mapped/*/*.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0af0d890",
   "metadata": {},
   "outputs": [],
   "source": [
    "featureCounts -T 23 -p -t exon -g gene_id -a ~/NAS_MEB/workspace/rnaseq/genome_ebi/gencode.v29.annotation.gtf -o ~/NAS_MEB/workspace/rnaseq/featurecounts.txt ~/NAS_MEB/workspace/rnaseq/mapped/*/*.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0efa63d7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.0\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m14 BAM files  \u001b[0m \u001b[0m                                  ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mfeaturecounts.txt\u001b[0m \u001b[0m                               ||\n",
      "||                 Summary : \u001b[36mfeaturecounts.txt.summary\u001b[0m \u001b[0m                       ||\n",
      "||              Annotation : \u001b[36mhg38.refGene.gtf (GTF)\u001b[0m \u001b[0m                          ||\n",
      "||      Dir for temp files : \u001b[36m/home/stud03/NAS_MEB/workspace/rnaseq/countT\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36myes\u001b[0m \u001b[0m                                             ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||          Chimeric reads : \u001b[36mcounted\u001b[0m \u001b[0m                                         ||\n",
      "||        Both ends mapped : \u001b[36mnot required\u001b[0m \u001b[0m                                    ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file hg38.refGene.gtf \u001b[0m... \u001b[0m                                 ||\n",
      "||    Features : \u001b[36m836541\u001b[0m \u001b[0m                                                      ||\n",
      "||    Meta-features : \u001b[36m28278\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m367\u001b[0m \u001b[0m                                              ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m16383290\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m10637360 (64.9%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m10.67 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m23298019\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m15773581 (67.7%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m15.88 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m17249142\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m10010591 (58.0%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m13.35 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m16135620\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m9501820 (58.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m10.17 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m22995282\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m12518114 (54.4%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m21.00 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m10630552\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m6342136 (59.7%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m7.58 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m9607522\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m5654417 (58.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m7.12 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m21347066\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m12339539 (57.8%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m21.30 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m11252174\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m6924745 (61.5%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m8.05 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m16249540\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m9332324 (57.4%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m17.27 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m14508316\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m9923534 (68.4%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m11.17 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m14171986\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m9242416 (65.2%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m9.44 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m26494372\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m16517102 (62.3%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m23.81 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m5695766\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m3621340 (63.6%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m3.70 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"/home/stud03/NAS_MEB/wo \u001b[0m ||\n",
      "|| rkspace/rnaseq/countTable2/featurecounts.txt.summary\" \u001b[0m                     ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#featureCounts -p -a ~/NAS_MEB/workspace/rnaseq/hg38.refGene.gtf -g gene_name -o ~/NAS_MEB/workspace/rnaseq/countTable2/featurecounts.txt ~/NAS_MEB/workspace/rnaseq/mapped/*/*.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c3735729",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RNA_NHEM_76_ACAGTG_L007_R1_001\n",
      "RNA_NHEM_76_ACAGTG_L007_R2_001\n",
      "NA_NHEM_77_GCCAAT_L007_R1_001\n",
      "RNA_NHEM_77_GCCAAT_L007_R2_001\n",
      "RNA_Sbcl2_64_TAGCTT_L007_R1_001\n",
      "RNA_Sbcl2_64_TAGCTT_L007_R2_001\n",
      "RNA_Sbcl2_70_TGACCA_L008_R1_001\n",
      "RNA_Sbcl2_70_TGACCA_L008_R2_001\n",
      "RNA_WM1158_68_CGATGT_L008_R1_001\n",
      "RNA_WM1158_68_CGATGT_L008_R2_001\n",
      "RNA_WM1158_74_ACTTGA_L007_R1_001\n",
      "RNA_WM1158_74_ACTTGA_L007_R2_001\n",
      "RNA_WM1366_66_CTTGTA_L007_R1_001\n",
      "RNA_WM1366_66_CTTGTA_L007_R2_001\n",
      "RNA_WM1366_72_GCCAAT_L008_R1_001\n",
      "RNA_WM1366_72_GCCAAT_L008_R2_001\n",
      "RNA_WM3211_65_GGCTAC_L007_R1_001\n",
      "RNA_WM3211_65_GGCTAC_L007_R2_001\n",
      "RNA_WM3211_71_ACAGTG_L008_R1_001\n",
      "RNA_WM3211_71_ACAGTG_L008_R2_001\n",
      "RNA_WM793_67_ATCACG_L008_R1_001\n",
      "RNA_WM793_67_ATCACG_L008_R2_001\n",
      "RNA_WM793_73_CAGATC_L008_R1_001\n",
      "RNA_WM793_73_CAGATC_L008_R2_001\n",
      "RNA_WM9_69_TTAGGC_L008_R1_001\n",
      "RNA_WM9_69_TTAGGC_L008_R2_001\n",
      "RNA_WM9_75_GATCAG_L007_R1_001\n",
      "RNA_WM9_75_GATCAG_L007_R2_001\n"
     ]
    }
   ],
   "source": [
    "cat SampleNames_fastq.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4fffa033",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7b58263d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.0\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m14 BAM files  \u001b[0m \u001b[0m                                  ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mfeaturecounts.txt\u001b[0m \u001b[0m                               ||\n",
      "||                 Summary : \u001b[36mfeaturecounts.txt.summary\u001b[0m \u001b[0m                       ||\n",
      "||              Annotation : \u001b[36mgencode.v29.annotation.gtf (GTF)\u001b[0m \u001b[0m                ||\n",
      "||      Dir for temp files : \u001b[36m/home/stud03/NAS_MEB/workspace/rnaseq/countT\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m23\u001b[0m \u001b[0m                                              ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36myes\u001b[0m \u001b[0m                                             ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||          Chimeric reads : \u001b[36mcounted\u001b[0m \u001b[0m                                         ||\n",
      "||        Both ends mapped : \u001b[36mnot required\u001b[0m \u001b[0m                                    ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file gencode.v29.annotation.gtf \u001b[0m... \u001b[0m                       ||\n",
      "||    Features : \u001b[36m1262773\u001b[0m \u001b[0m                                                     ||\n",
      "||    Meta-features : \u001b[36m57133\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m25\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m16383290\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m9977764 (60.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m8.04 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m23298019\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m14718249 (63.2%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m11.94 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m17249142\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m9135840 (53.0%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m9.76 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m16135620\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m8965458 (55.6%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m7.67 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m22995282\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m11520597 (50.1%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m15.22 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m10630552\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m5882949 (55.3%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m5.29 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m9607522\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m5295638 (55.1%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m5.26 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m21347066\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m11285996 (52.9%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m15.51 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m11252174\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m6497136 (57.7%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m6.15 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m16249540\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m8475523 (52.2%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m12.07 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m14508316\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m9142628 (63.0%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m7.68 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m14171986\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m8559477 (60.4%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m6.81 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m26494372\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m14454859 (54.6%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m17.30 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m5695766\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m3247935 (57.0%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m2.89 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"/home/stud03/NAS_MEB/wo \u001b[0m ||\n",
      "|| rkspace/rnaseq/countTable3/featurecounts.txt.summary\" \u001b[0m                     ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "featureCounts -T 23 -p  -g gene_name -a ~/NAS_MEB/workspace/rnaseq/genome_ebi/gencode.v29.annotation.gtf -o ~/NAS_MEB/workspace/rnaseq/countTable3/featurecounts.txt ~/NAS_MEB/workspace/rnaseq/mapped/*/*.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c2893809",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.0\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m14 BAM files  \u001b[0m \u001b[0m                                  ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||                           \u001b[32mo\u001b[36m Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m                 ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mfeaturecounts.txt\u001b[0m \u001b[0m                               ||\n",
      "||                 Summary : \u001b[36mfeaturecounts.txt.summary\u001b[0m \u001b[0m                       ||\n",
      "||              Annotation : \u001b[36mgencode.v29.annotation.gtf (GTF)\u001b[0m \u001b[0m                ||\n",
      "||      Dir for temp files : \u001b[36m/home/stud03/NAS_MEB/workspace/rnaseq/countT\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m23\u001b[0m \u001b[0m                                              ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36myes\u001b[0m \u001b[0m                                             ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||          Chimeric reads : \u001b[36mcounted\u001b[0m \u001b[0m                                         ||\n",
      "||        Both ends mapped : \u001b[36mnot required\u001b[0m \u001b[0m                                    ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file gencode.v29.annotation.gtf \u001b[0m... \u001b[0m                       ||\n",
      "||    Features : \u001b[36m1262773\u001b[0m \u001b[0m                                                     ||\n",
      "||    Meta-features : \u001b[36m58721\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m25\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m16383290\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m9920772 (60.6%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m8.20 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m23298019\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m14628815 (62.8%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m12.22 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m17249142\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m9102556 (52.8%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m9.82 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m16135620\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m8933548 (55.4%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m7.59 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m22995282\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m11427146 (49.7%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m14.89 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m10630552\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m5838804 (54.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m5.41 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m9607522\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m5276454 (54.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m5.30 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m21347066\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m11218527 (52.6%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m15.19 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m11252174\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m6467365 (57.5%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m5.97 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m16249540\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m8412118 (51.8%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m12.06 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m14508316\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m9111122 (62.8%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m7.66 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m14171986\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m8528542 (60.2%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m6.76 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m26494372\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m14340018 (54.1%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m17.04 minutes\u001b[0m \u001b[0m                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Aligned.sortedByCoord.out.bam... \u001b[0m                         ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m5695766\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m3221739 (56.6%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m2.78 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"/home/stud03/NAS_MEB/wo \u001b[0m ||\n",
      "|| rkspace/rnaseq/countTable4/featurecounts.txt.summary\" \u001b[0m                     ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "featureCounts -T 23 -p -t exon -g gene_id -a ~/NAS_MEB/workspace/rnaseq/genome_ebi/gencode.v29.annotation.gtf -o ~/NAS_MEB/workspace/rnaseq/countTable4/featurecounts.txt ~/NAS_MEB/workspace/rnaseq/mapped/*/*.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "757058b0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/stud03/workspace/rnaseq/RNA_REFS_DIR\n"
     ]
    }
   ],
   "source": [
    "~/workspace/rnaseq/RNA_REFS_DIR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "872edf9d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Master_thesis.ipynb  Untitled.ipynb  hg38.fa  hg_ucsc.gtf\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fc09011d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# now it works.. limiting ram to 24GB\n",
    "# https://github.com/alexdobin/STAR/issues/266\n",
    "STAR --runMode genomeGenerate --runThreadN 14 --genomeDir ~/workspace/rnaseq/geneIndex --genomeFastaFiles ~/workspace/rnaseq/RNA_REFS_DIR/hg38.fa --sjdbGTFfile ~/workspace/rnaseq/RNA_REFS_DIR/hg_ucsc.gtf --sjdbOverhang 99 --limitGenomeGenerateRAM 24000000000 --genomeSAsparseD 2\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "67bf4746",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd workspace/rnaseq/geneIndex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c468772d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genome   chrLength.txt      exonGeTrInfo.tab      sjdbInfo.txt\n",
      "Log.out  chrName.txt        exonInfo.tab          sjdbList.fromGTF.out.tab\n",
      "SA       chrNameLength.txt  geneInfo.tab          sjdbList.out.tab\n",
      "SAindex  chrStart.txt       genomeParameters.txt  transcriptInfo.tab\n"
     ]
    }
   ],
   "source": [
    "ls\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "31589861",
   "metadata": {},
   "outputs": [],
   "source": [
    "STAR --runMode alignReads --genomeDir "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ce2986b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd workspace/rnaseq/trimData"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f1d2d187",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "gzip: RNA_NHEM_76_trimmed_1.fastq: No space left on device\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "gunzip -d *.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a6af65a9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "              total        used        free      shared  buff/cache   available\n",
      "Mem:             31           2          19           0           8          28\n",
      "Swap:             0           0           0\n"
     ]
    }
   ],
   "source": [
    "free -g"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c3cbcd08",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filesystem                              Size  Used Avail Use% Mounted on\n",
      "udev                                     16G     0   16G   0% /dev\n",
      "tmpfs                                   3.2G  2.9M  3.2G   1% /run\n",
      "/dev/vda1                               146G  145G  1.1G 100% /\n",
      "tmpfs                                    16G     0   16G   0% /dev/shm\n",
      "tmpfs                                   5.0M     0  5.0M   0% /run/lock\n",
      "tmpfs                                    16G     0   16G   0% /sys/fs/cgroup\n",
      "/dev/loop1                               56M   56M     0 100% /snap/core18/2253\n",
      "/dev/loop4                               33M   33M     0 100% /snap/ruby/242\n",
      "/dev/loop5                               68M   68M     0 100% /snap/lxd/21803\n",
      "/dev/vda15                              105M  5.2M  100M   5% /boot/efi\n",
      "/dev/loop6                               68M   68M     0 100% /snap/lxd/21835\n",
      "//195.37.232.90/hpc_storage              29T  9.6T   19T  35% /home/ubuntu/NAS\n",
      "/dev/loop10                              62M   62M     0 100% /snap/core20/1270\n",
      "/dev/loop2                               44M   44M     0 100% /snap/snapd/14295\n",
      "//195.37.232.90/hpc_storage/01_stefanf   29T  9.6T   19T  35% /home/sf/NAS_LSI\n",
      "tmpfs                                   3.2G     0  3.2G   0% /run/user/1001\n",
      "tmpfs                                   3.2G     0  3.2G   0% /run/user/1002\n",
      "/dev/loop8                               35M   35M     0 100% /snap/ruby/247\n",
      "tmpfs                                   3.2G  4.0K  3.2G   1% /run/user/1004\n",
      "/dev/loop7                               56M   56M     0 100% /snap/core18/2284\n",
      "/dev/loop0                               44M   44M     0 100% /snap/snapd/14549\n",
      "/dev/loop11                              62M   62M     0 100% /snap/core20/1328\n"
     ]
    }
   ],
   "source": [
    "df -h\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2bf3b461",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Alignment "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f69a7ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "STAR --runMode alignReads --genomeDir ~/NAS_MEB/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "adf59caf",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/stud03/NAS_MEB/trimData\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8ef5c3c",
   "metadata": {},
   "outputs": [],
   "source": [
    "STAR --runMode alignReads --genomeDir ~/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn {file path} --runThreadN 10 --outFileNamePrefix ~/path product"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "beccac0e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#loop to do it once "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "73a5e224",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_NHEM_76_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_NHEM_76_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 20 21:35:59 ..... started STAR run\n",
      "Feb 20 21:35:59 ..... loading genome\n",
      "Feb 20 21:36:39 ..... started mapping\n",
      "Feb 20 21:42:57 ..... finished mapping\n",
      "Feb 20 21:44:15 ..... started sorting BAM\n",
      "Feb 20 21:46:33 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_NHEM_76_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_NHEM_76_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 20 21:46:35 ..... started STAR run\n",
      "Feb 20 21:46:35 ..... loading genome\n",
      "Feb 20 21:47:03 ..... started mapping\n",
      "Feb 20 21:53:39 ..... finished mapping\n",
      "Feb 20 21:55:06 ..... started sorting BAM\n",
      "Feb 20 21:57:35 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_NHEM_77_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_NHEM_77_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 20 21:57:38 ..... started STAR run\n",
      "Feb 20 21:57:38 ..... loading genome\n",
      "Feb 20 21:57:58 ..... started mapping\n",
      "Feb 20 22:07:26 ..... finished mapping\n",
      "Feb 20 22:08:41 ..... started sorting BAM\n",
      "Feb 20 22:14:51 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_NHEM_77_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_NHEM_77_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 20 22:14:54 ..... started STAR run\n",
      "Feb 20 22:14:54 ..... loading genome\n",
      "Feb 20 22:15:24 ..... started mapping\n",
      "Killed\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_Sbcl2_64_TAGCTT_L007_R1_001.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_Sbcl2_64_TAGCTT_L007_R1_001.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 20 22:17:51 ..... started STAR run\n",
      "Feb 20 22:17:51 ..... loading genome\n",
      "Feb 20 22:18:34 ..... started mapping\n",
      "Feb 20 22:26:44 ..... finished mapping\n",
      "Feb 20 22:28:25 ..... started sorting BAM\n",
      "Feb 20 22:34:11 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_Sbcl2_64_TAGCTT_L007_R2_001.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_Sbcl2_64_TAGCTT_L007_R2_001.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 20 22:34:14 ..... started STAR run\n",
      "Feb 20 22:34:14 ..... loading genome\n",
      "Feb 20 22:34:53 ..... started mapping\n",
      "Feb 20 22:43:20 ..... finished mapping\n",
      "Feb 20 22:44:42 ..... started sorting BAM\n",
      "Feb 20 22:51:08 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_Sbcl2_64_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_Sbcl2_64_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 20 22:51:11 ..... started STAR run\n",
      "Feb 20 22:51:11 ..... loading genome\n",
      "Feb 20 22:51:37 ..... started mapping\n",
      "Feb 20 22:59:02 ..... finished mapping\n",
      "Feb 20 23:00:31 ..... started sorting BAM\n",
      "Feb 20 23:06:37 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_Sbcl2_64_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_Sbcl2_64_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 20 23:06:40 ..... started STAR run\n",
      "Feb 20 23:06:40 ..... loading genome\n",
      "Feb 20 23:07:00 ..... started mapping\n",
      "Feb 20 23:14:50 ..... finished mapping\n",
      "Feb 20 23:16:11 ..... started sorting BAM\n",
      "Feb 20 23:22:05 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_Sbcl2_70_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_Sbcl2_70_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 20 23:22:08 ..... started STAR run\n",
      "Feb 20 23:22:08 ..... loading genome\n",
      "Feb 20 23:22:30 ..... started mapping\n",
      "Killed\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_Sbcl2_70_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_Sbcl2_70_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 20 23:28:52 ..... started STAR run\n",
      "Feb 20 23:28:52 ..... loading genome\n",
      "Killed\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1158_68_CGATGT_L008_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM1158_68_CGATGT_L008_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 20 23:29:33 ..... started STAR run\n",
      "Feb 20 23:29:33 ..... loading genome\n",
      "Killed\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1158_68_CGATGT_L008_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM1158_68_CGATGT_L008_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 20 23:30:11 ..... started STAR run\n",
      "Feb 20 23:30:11 ..... loading genome\n",
      "Feb 20 23:30:53 ..... started mapping\n",
      "Feb 20 23:40:45 ..... finished mapping\n",
      "Feb 20 23:42:04 ..... started sorting BAM\n",
      "Feb 20 23:50:25 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1158_74_Trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM1158_74_Trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 20 23:50:28 ..... started STAR run\n",
      "Feb 20 23:50:28 ..... loading genome\n",
      "Feb 20 23:51:07 ..... started mapping\n",
      "Feb 20 23:56:05 ..... finished mapping\n",
      "Feb 20 23:57:47 ..... started sorting BAM\n",
      "Feb 21 00:00:10 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1158_74_Trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM1158_74_Trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 00:00:12 ..... started STAR run\n",
      "Feb 21 00:00:12 ..... loading genome\n",
      "Feb 21 00:00:40 ..... started mapping\n",
      "Feb 21 00:05:24 ..... finished mapping\n",
      "Feb 21 00:06:36 ..... started sorting BAM\n",
      "Feb 21 00:09:20 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1366_66_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM1366_66_trimmed_1.fastq\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 00:09:23 ..... started STAR run\n",
      "Feb 21 00:09:23 ..... loading genome\n",
      "Feb 21 00:09:44 ..... started mapping\n",
      "Feb 21 00:14:02 ..... finished mapping\n",
      "Feb 21 00:15:05 ..... started sorting BAM\n",
      "Feb 21 00:16:44 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1366_66_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM1366_66_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 00:16:47 ..... started STAR run\n",
      "Feb 21 00:16:47 ..... loading genome\n",
      "Feb 21 00:17:08 ..... started mapping\n",
      "Feb 21 00:22:13 ..... finished mapping\n",
      "Feb 21 00:23:25 ..... started sorting BAM\n",
      "Feb 21 00:25:07 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1366_72_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM1366_72_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 00:25:09 ..... started STAR run\n",
      "Feb 21 00:25:09 ..... loading genome\n",
      "Killed\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1366_72_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM1366_72_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 00:25:48 ..... started STAR run\n",
      "Feb 21 00:25:48 ..... loading genome\n",
      "Feb 21 00:26:33 ..... started mapping\n",
      "Feb 21 00:36:35 ..... finished mapping\n",
      "Feb 21 00:38:03 ..... started sorting BAM\n",
      "Feb 21 00:41:40 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM3211_65_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM3211_65_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 00:41:43 ..... started STAR run\n",
      "Feb 21 00:41:43 ..... loading genome\n",
      "Feb 21 00:42:22 ..... started mapping\n",
      "Feb 21 00:47:36 ..... finished mapping\n",
      "Feb 21 00:48:49 ..... started sorting BAM\n",
      "Feb 21 00:50:25 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM3211_65_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM3211_65_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 00:50:28 ..... started STAR run\n",
      "Feb 21 00:50:28 ..... loading genome\n",
      "Feb 21 00:50:55 ..... started mapping\n",
      "Feb 21 00:56:12 ..... finished mapping\n",
      "Feb 21 00:57:29 ..... started sorting BAM\n",
      "Feb 21 00:59:16 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM3211_71_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM3211_71_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 00:59:19 ..... started STAR run\n",
      "Feb 21 00:59:19 ..... loading genome\n",
      "Feb 21 00:59:41 ..... started mapping\n",
      "Feb 21 01:06:53 ..... finished mapping\n",
      "Feb 21 01:08:08 ..... started sorting BAM\n",
      "Feb 21 01:10:23 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM3211_71_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM3211_71_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 01:10:26 ..... started STAR run\n",
      "Feb 21 01:10:26 ..... loading genome\n",
      "Feb 21 01:10:47 ..... started mapping\n",
      "Feb 21 01:18:06 ..... finished mapping\n",
      "Feb 21 01:19:30 ..... started sorting BAM\n",
      "Feb 21 01:22:10 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM793_67_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM793_67_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 01:22:14 ..... started STAR run\n",
      "Feb 21 01:22:14 ..... loading genome\n",
      "Feb 21 01:22:35 ..... started mapping\n",
      "Feb 21 01:29:13 ..... finished mapping\n",
      "Feb 21 01:30:28 ..... started sorting BAM\n",
      "Feb 21 01:32:35 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM793_67_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM793_67_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 01:32:38 ..... started STAR run\n",
      "Feb 21 01:32:38 ..... loading genome\n",
      "Feb 21 01:32:59 ..... started mapping\n",
      "Feb 21 01:39:35 ..... finished mapping\n",
      "Feb 21 01:40:51 ..... started sorting BAM\n",
      "Feb 21 01:43:03 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM793_73_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM793_73_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 01:43:06 ..... started STAR run\n",
      "Feb 21 01:43:06 ..... loading genome\n",
      "Feb 21 01:43:30 ..... started mapping\n",
      "Feb 21 01:51:28 ..... finished mapping\n",
      "Feb 21 01:52:41 ..... started sorting BAM\n",
      "Feb 21 01:54:46 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM793_73_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM793_73_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 01:54:49 ..... started STAR run\n",
      "Feb 21 01:54:49 ..... loading genome\n",
      "Feb 21 01:55:09 ..... started mapping\n",
      "Feb 21 02:03:01 ..... finished mapping\n",
      "Feb 21 02:04:36 ..... started sorting BAM\n",
      "Feb 21 02:06:55 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM9_69_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM9_69_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 02:06:57 ..... started STAR run\n",
      "Feb 21 02:06:57 ..... loading genome\n",
      "Feb 21 02:07:21 ..... started mapping\n",
      "Feb 21 02:18:04 ..... finished mapping\n",
      "Feb 21 02:19:45 ..... started sorting BAM\n",
      "Feb 21 02:23:33 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM9_69_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM9_69_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 02:23:35 ..... started STAR run\n",
      "Feb 21 02:23:35 ..... loading genome\n",
      "Feb 21 02:23:56 ..... started mapping\n",
      "Feb 21 02:34:54 ..... finished mapping\n",
      "Feb 21 02:36:40 ..... started sorting BAM\n",
      "Feb 21 02:41:09 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM9_75_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM9_75_trimmed_1.fastq\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 02:41:12 ..... started STAR run\n",
      "Feb 21 02:41:12 ..... loading genome\n",
      "Feb 21 02:41:34 ..... started mapping\n",
      "Feb 21 02:45:55 ..... finished mapping\n",
      "Feb 21 02:47:20 ..... started sorting BAM\n",
      "Feb 21 02:48:13 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM9_75_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mappedRNA_WM9_75_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Feb 21 02:48:15 ..... started STAR run\n",
      "Feb 21 02:48:15 ..... loading genome\n",
      "Feb 21 02:48:36 ..... started mapping\n",
      "Feb 21 02:53:05 ..... finished mapping\n",
      "Feb 21 02:54:10 ..... started sorting BAM\n",
      "Feb 21 02:55:04 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "#for file in *.fastq; do STAR --runMode alignReads --genomeDir ~/workspace/rnaseq//geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn ${file} --runThreadN 14 --outFileNamePrefix ~/NAS_MEB/mapped${file}; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c055f829",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mRNA_NHEM_76_ACAGTG_L007_R1_001.fastq\u001b[0m     \u001b[01;32mRNA_WM1366_72_trimmed_1.fastq\u001b[0m\n",
      "\u001b[01;32mRNA_NHEM_76_ACAGTG_L007_R2_001.fastq\u001b[0m     \u001b[01;32mRNA_WM1366_72_trimmed_2.fastq\u001b[0m\n",
      "\u001b[01;32mRNA_NHEM_77_GCCAAT_L007_R1_001.fastq\u001b[0m     \u001b[01;32mRNA_WM3211_65_trimmed.log\u001b[0m\n",
      "\u001b[01;32mRNA_NHEM_77_GCCAAT_L007_R2_001.fastq\u001b[0m     \u001b[01;32mRNA_WM3211_65_trimmed_1.fastq\u001b[0m\n",
      "\u001b[01;32mRNA_Sbcl2_64_trimmed.log\u001b[0m                 \u001b[01;32mRNA_WM3211_65_trimmed_2.fastq\u001b[0m\n",
      "\u001b[01;32mRNA_Sbcl2_64_trimmed_1.fastq\u001b[0m             \u001b[01;32mRNA_WM3211_71_trimmed.log\u001b[0m\n",
      "\u001b[01;32mRNA_Sbcl2_64_trimmed_2.fastq\u001b[0m             \u001b[01;32mRNA_WM3211_71_trimmed_1.fastq\u001b[0m\n",
      "\u001b[01;32mRNA_Sbcl2_70_trimmed.log\u001b[0m                 \u001b[01;32mRNA_WM3211_71_trimmed_2.fastq\u001b[0m\n",
      "\u001b[01;32mRNA_Sbcl2_70_trimmed_1.fastq\u001b[0m             \u001b[01;32mRNA_WM793_67_ATCACG_L008_R1_001.fastq\u001b[0m\n",
      "\u001b[01;32mRNA_Sbcl2_70_trimmed_2.fastq\u001b[0m             \u001b[01;32mRNA_WM793_67_ATCACG_L008_R2_001.fastq\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_68_CGATGT_L008.log\u001b[0m            \u001b[01;32mRNA_WM793_73_trimmed.log\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_68_CGATGT_L008_1.fastq\u001b[0m        \u001b[01;32mRNA_WM793_73_trimmed_1.fastq\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_68_CGATGT_L008_1_fastqc.html\u001b[0m  \u001b[01;32mRNA_WM793_73_trimmed_2.fastq\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_68_CGATGT_L008_1_fastqc.zip\u001b[0m   \u001b[01;32mRNA_WM9_75_trimmed.log\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_68_CGATGT_L008_2.fastq\u001b[0m        \u001b[01;32mRNA_WM9_75_trimmed_1.fastq\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_68_CGATGT_L008_2_fastqc.html\u001b[0m  \u001b[01;32mRNA_WM9_75_trimmed_2.fastq\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_68_CGATGT_L008_2_fastqc.zip\u001b[0m   \u001b[01;32mTruSeq2-PE.fa\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_74_Trimmed.log\u001b[0m                \u001b[01;32mTruSeq3-PE.fa\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_74_Trimmed_1.fastq\u001b[0m            \u001b[01;34madapters\u001b[0m\n",
      "\u001b[01;32mRNA_WM1158_74_Trimmed_2.fastq\u001b[0m            \u001b[01;32millumina_mine.fa\u001b[0m\n",
      "\u001b[01;32mRNA_WM1366_66_trimmed.log\u001b[0m                \u001b[01;32millumina_multiplex.fa\u001b[0m\n",
      "\u001b[01;32mRNA_WM1366_66_trimmed_1.fastq\u001b[0m            \u001b[01;34msame_qc\u001b[0m\n",
      "\u001b[01;32mRNA_WM1366_66_trimmed_2.fastq\u001b[0m            \u001b[01;32msampleNames.txt\u001b[0m\n",
      "\u001b[01;32mRNA_WM1366_72_trimmed.log\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be22eaa8",
   "metadata": {},
   "source": [
    "STAR --runMode alignReads --genomeDir ~/NAS_MEB/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn ${file} --runThreadN 14 --outFileNamePrefix ~/NAS_MEB/mapped${file}; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7a369853",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_NHEM_76_ACAGTG_L007_R1_001.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_NHEM_76_ACAGTG_L007_R1_001.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 07:02:34 ..... started STAR run\n",
      "Mar 05 07:02:34 ..... loading genome\n",
      "Mar 05 07:03:13 ..... started mapping\n",
      "Mar 05 07:10:31 ..... finished mapping\n",
      "Mar 05 07:11:54 ..... started sorting BAM\n",
      "Mar 05 07:14:21 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_NHEM_76_ACAGTG_L007_R2_001.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_NHEM_76_ACAGTG_L007_R2_001.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 07:14:23 ..... started STAR run\n",
      "Mar 05 07:14:24 ..... loading genome\n",
      "Mar 05 07:14:53 ..... started mapping\n",
      "Mar 05 07:22:19 ..... finished mapping\n",
      "Mar 05 07:23:35 ..... started sorting BAM\n",
      "Mar 05 07:26:20 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_NHEM_77_GCCAAT_L007_R1_001.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_NHEM_77_GCCAAT_L007_R1_001.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 07:26:23 ..... started STAR run\n",
      "Mar 05 07:26:23 ..... loading genome\n",
      "Mar 05 07:26:44 ..... started mapping\n",
      "Mar 05 07:37:52 ..... finished mapping\n",
      "Mar 05 07:39:34 ..... started sorting BAM\n",
      "Mar 05 07:43:12 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_NHEM_77_GCCAAT_L007_R2_001.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_NHEM_77_GCCAAT_L007_R2_001.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 07:43:15 ..... started STAR run\n",
      "Mar 05 07:43:15 ..... loading genome\n",
      "Mar 05 07:43:39 ..... started mapping\n",
      "Mar 05 07:54:59 ..... finished mapping\n",
      "Mar 05 07:56:22 ..... started sorting BAM\n",
      "Mar 05 08:00:18 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_Sbcl2_64_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_Sbcl2_64_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 08:00:20 ..... started STAR run\n",
      "Mar 05 08:00:20 ..... loading genome\n",
      "Mar 05 08:00:42 ..... started mapping\n",
      "Mar 05 08:08:13 ..... finished mapping\n",
      "Mar 05 08:09:49 ..... started sorting BAM\n",
      "Mar 05 08:12:25 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_Sbcl2_64_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_Sbcl2_64_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 08:12:28 ..... started STAR run\n",
      "Mar 05 08:12:28 ..... loading genome\n",
      "Mar 05 08:12:54 ..... started mapping\n",
      "Mar 05 08:20:33 ..... finished mapping\n",
      "Mar 05 08:22:05 ..... started sorting BAM\n",
      "Mar 05 08:24:55 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_Sbcl2_70_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_Sbcl2_70_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 08:24:58 ..... started STAR run\n",
      "Mar 05 08:24:58 ..... loading genome\n",
      "Mar 05 08:25:19 ..... started mapping\n",
      "Mar 05 08:34:36 ..... finished mapping\n",
      "Mar 05 08:35:56 ..... started sorting BAM\n",
      "Mar 05 08:38:29 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_Sbcl2_70_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_Sbcl2_70_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 08:38:31 ..... started STAR run\n",
      "Mar 05 08:38:31 ..... loading genome\n",
      "Mar 05 08:38:54 ..... started mapping\n",
      "Mar 05 08:48:22 ..... finished mapping\n",
      "Mar 05 08:49:37 ..... started sorting BAM\n",
      "Mar 05 08:52:22 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1158_68_CGATGT_L008_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM1158_68_CGATGT_L008_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 08:52:24 ..... started STAR run\n",
      "Mar 05 08:52:24 ..... loading genome\n",
      "Mar 05 08:52:45 ..... started mapping\n",
      "Mar 05 09:02:41 ..... finished mapping\n",
      "Mar 05 09:04:08 ..... started sorting BAM\n",
      "Mar 05 09:07:39 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1158_68_CGATGT_L008_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM1158_68_CGATGT_L008_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 09:07:42 ..... started STAR run\n",
      "Mar 05 09:07:42 ..... loading genome\n",
      "Mar 05 09:08:05 ..... started mapping\n",
      "Mar 05 09:18:09 ..... finished mapping\n",
      "Mar 05 09:19:48 ..... started sorting BAM\n",
      "Mar 05 09:23:43 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1158_74_Trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM1158_74_Trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 09:23:46 ..... started STAR run\n",
      "Mar 05 09:23:46 ..... loading genome\n",
      "Mar 05 09:24:07 ..... started mapping\n",
      "Mar 05 09:29:06 ..... finished mapping\n",
      "Mar 05 09:30:20 ..... started sorting BAM\n",
      "Mar 05 09:32:00 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1158_74_Trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM1158_74_Trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 09:32:02 ..... started STAR run\n",
      "Mar 05 09:32:02 ..... loading genome\n",
      "Mar 05 09:32:25 ..... started mapping\n",
      "Mar 05 09:37:28 ..... finished mapping\n",
      "Mar 05 09:38:52 ..... started sorting BAM\n",
      "Mar 05 09:40:37 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1366_66_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM1366_66_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 09:40:39 ..... started STAR run\n",
      "Mar 05 09:40:39 ..... loading genome\n",
      "Mar 05 09:41:00 ..... started mapping\n",
      "Mar 05 09:45:29 ..... finished mapping\n",
      "Mar 05 09:46:35 ..... started sorting BAM\n",
      "Mar 05 09:48:00 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1366_66_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM1366_66_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mar 05 09:48:02 ..... started STAR run\n",
      "Mar 05 09:48:02 ..... loading genome\n",
      "Mar 05 09:48:25 ..... started mapping\n",
      "Mar 05 09:52:54 ..... finished mapping\n",
      "Mar 05 09:53:51 ..... started sorting BAM\n",
      "Mar 05 09:55:27 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1366_72_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM1366_72_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 09:55:29 ..... started STAR run\n",
      "Mar 05 09:55:29 ..... loading genome\n",
      "Mar 05 09:55:50 ..... started mapping\n",
      "Mar 05 10:05:11 ..... finished mapping\n",
      "Mar 05 10:06:40 ..... started sorting BAM\n",
      "Mar 05 10:09:56 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM1366_72_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM1366_72_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 10:09:59 ..... started STAR run\n",
      "Mar 05 10:09:59 ..... loading genome\n",
      "Mar 05 10:10:35 ..... started mapping\n",
      "Mar 05 10:20:26 ..... finished mapping\n",
      "Mar 05 10:21:59 ..... started sorting BAM\n",
      "Mar 05 10:25:36 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM3211_65_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM3211_65_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 10:25:39 ..... started STAR run\n",
      "Mar 05 10:25:39 ..... loading genome\n",
      "Mar 05 10:26:03 ..... started mapping\n",
      "Mar 05 10:31:34 ..... finished mapping\n",
      "Mar 05 10:32:45 ..... started sorting BAM\n",
      "Mar 05 10:34:25 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM3211_65_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM3211_65_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 10:34:28 ..... started STAR run\n",
      "Mar 05 10:34:28 ..... loading genome\n",
      "Mar 05 10:34:48 ..... started mapping\n",
      "Mar 05 10:40:19 ..... finished mapping\n",
      "Mar 05 10:41:27 ..... started sorting BAM\n",
      "Mar 05 10:43:22 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM3211_71_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM3211_71_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 10:43:24 ..... started STAR run\n",
      "Mar 05 10:43:24 ..... loading genome\n",
      "Mar 05 10:43:50 ..... started mapping\n",
      "Mar 05 10:51:03 ..... finished mapping\n",
      "Mar 05 10:52:28 ..... started sorting BAM\n",
      "Mar 05 10:54:52 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM3211_71_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM3211_71_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 10:54:54 ..... started STAR run\n",
      "Mar 05 10:54:54 ..... loading genome\n",
      "Mar 05 10:55:14 ..... started mapping\n",
      "Mar 05 11:02:50 ..... finished mapping\n",
      "Mar 05 11:03:55 ..... started sorting BAM\n",
      "Mar 05 11:06:43 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM793_67_ATCACG_L008_R1_001.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM793_67_ATCACG_L008_R1_001.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 11:06:45 ..... started STAR run\n",
      "Mar 05 11:06:45 ..... loading genome\n",
      "Mar 05 11:07:08 ..... started mapping\n",
      "Mar 05 11:14:49 ..... finished mapping\n",
      "Mar 05 11:16:14 ..... started sorting BAM\n",
      "Mar 05 11:18:36 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM793_67_ATCACG_L008_R2_001.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM793_67_ATCACG_L008_R2_001.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 11:18:38 ..... started STAR run\n",
      "Mar 05 11:18:39 ..... loading genome\n",
      "Mar 05 11:18:59 ..... started mapping\n",
      "Mar 05 11:26:35 ..... finished mapping\n",
      "Mar 05 11:27:58 ..... started sorting BAM\n",
      "Mar 05 11:30:33 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM793_73_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM793_73_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 11:30:35 ..... started STAR run\n",
      "Mar 05 11:30:36 ..... loading genome\n",
      "Mar 05 11:30:59 ..... started mapping\n",
      "Mar 05 11:39:14 ..... finished mapping\n",
      "Mar 05 11:40:27 ..... started sorting BAM\n",
      "Mar 05 11:42:40 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM793_73_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM793_73_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 11:42:42 ..... started STAR run\n",
      "Mar 05 11:42:42 ..... loading genome\n",
      "Mar 05 11:43:03 ..... started mapping\n",
      "Mar 05 11:51:25 ..... finished mapping\n",
      "Mar 05 11:52:45 ..... started sorting BAM\n",
      "Mar 05 11:55:10 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM9_75_trimmed_1.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM9_75_trimmed_1.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 11:55:12 ..... started STAR run\n",
      "Mar 05 11:55:12 ..... loading genome\n",
      "Mar 05 11:55:35 ..... started mapping\n",
      "Mar 05 11:59:44 ..... finished mapping\n",
      "Mar 05 12:00:46 ..... started sorting BAM\n",
      "Mar 05 12:01:37 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM9_75_trimmed_2.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM9_75_trimmed_2.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 05 12:01:39 ..... started STAR run\n",
      "Mar 05 12:01:39 ..... loading genome\n",
      "Mar 05 12:02:01 ..... started mapping\n",
      "Mar 05 12:06:10 ..... finished mapping\n",
      "Mar 05 12:07:11 ..... started sorting BAM\n",
      "Mar 05 12:08:02 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "#for file in *.fastq; do STAR --runMode alignReads --genomeDir ~/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn ${file} --runThreadN 14 --outFileNamePrefix ~/NAS_MEB/mapped2${file}; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b118013d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM9_69_TTAGGC_L008_R1_001.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM9_69_TTAGGC_L008_R1_001.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 06 09:16:04 ..... started STAR run\n",
      "Mar 06 09:16:04 ..... loading genome\n",
      "Mar 06 09:16:27 ..... started mapping\n",
      "Mar 06 09:28:37 ..... finished mapping\n",
      "Mar 06 09:30:23 ..... started sorting BAM\n",
      "Mar 06 09:34:25 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --genomeDir /home/stud03/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn RNA_WM9_69_TTAGGC_L008_R2_001.fastq --runThreadN 14 --outFileNamePrefix /home/stud03/NAS_MEB/mapped2RNA_WM9_69_TTAGGC_L008_R2_001.fastq\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 06 09:34:28 ..... started STAR run\n",
      "Mar 06 09:34:28 ..... loading genome\n",
      "Mar 06 09:34:50 ..... started mapping\n",
      "Mar 06 09:47:19 ..... finished mapping\n",
      "Mar 06 09:48:39 ..... started sorting BAM\n",
      "Mar 06 09:53:11 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "#for file in RNA_WM9_69_TTAGGC_L008*.fastq; do STAR --runMode alignReads --genomeDir ~/workspace/rnaseq/geneIndex/ --outSAMtype BAM SortedByCoordinate --readFilesIn ${file} --runThreadN 14 --outFileNamePrefix ~/NAS_MEB/mapped2${file}; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "832bb418",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd NAS_MEB/trimData"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fa2e9700",
   "metadata": {},
   "source": [
    "#### Count table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "da1d376c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.0\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m28 BAM files  \u001b[0m \u001b[0m                                  ||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_NHEM_76_ACAGTG_L007_R1_001.fast\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_NHEM_76_ACAGTG_L007_R2_001.fast\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_NHEM_77_GCCAAT_L007_R1_001.fast\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_NHEM_77_GCCAAT_L007_R2_001.fast\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_Sbcl2_64_trimmed_1.fastqAligned\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_Sbcl2_64_trimmed_2.fastqAligned\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_Sbcl2_70_trimmed_1.fastqAligned\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_Sbcl2_70_trimmed_2.fastqAligned\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM1158_68_CGATGT_L008_1.fastqAl\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM1158_68_CGATGT_L008_2.fastqAl\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM1158_74_Trimmed_1.fastqAligne\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM1158_74_Trimmed_2.fastqAligne\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM1366_66_trimmed_1.fastqAligne\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM1366_66_trimmed_2.fastqAligne\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM1366_72_trimmed_1.fastqAligne\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM1366_72_trimmed_2.fastqAligne\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM3211_65_trimmed_1.fastqAligne\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM3211_65_trimmed_2.fastqAligne\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM3211_71_trimmed_1.fastqAligne\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM3211_71_trimmed_2.fastqAligne\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM793_67_ATCACG_L008_R1_001.fas\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM793_67_ATCACG_L008_R2_001.fas\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM793_73_trimmed_1.fastqAligned\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM793_73_trimmed_2.fastqAligned\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM9_69_TTAGGC_L008_R1_001.fastq\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM9_69_TTAGGC_L008_R2_001.fastq\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM9_75_trimmed_1.fastqAligned.s\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m mapped2RNA_WM9_75_trimmed_2.fastqAligned.s\u001b[0m ... \u001b[0m||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mfeaturecounts.txt\u001b[0m \u001b[0m                               ||\n",
      "||                 Summary : \u001b[36mfeaturecounts.txt.summary\u001b[0m \u001b[0m                       ||\n",
      "||              Annotation : \u001b[36mhg_ucsc.gtf (GTF)\u001b[0m \u001b[0m                               ||\n",
      "||      Dir for temp files : \u001b[36m/home/stud03/NAS_MEB/countTable\u001b[0m \u001b[0m                 ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m8\u001b[0m \u001b[0m                                               ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36myes\u001b[0m \u001b[0m                                             ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||          Chimeric reads : \u001b[36mcounted\u001b[0m \u001b[0m                                         ||\n",
      "||        Both ends mapped : \u001b[36mnot required\u001b[0m \u001b[0m                                    ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file hg_ucsc.gtf \u001b[0m... \u001b[0m                                      ||\n",
      "||    Features : \u001b[36m896726\u001b[0m \u001b[0m                                                      ||\n",
      "||    Meta-features : \u001b[36m76501\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m369\u001b[0m \u001b[0m                                              ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_NHEM_76_ACAGTG_L007_R1_001.fastqAligned.so\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m18623878\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m3821723 (20.5%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m1.99 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_NHEM_76_ACAGTG_L007_R2_001.fastqAligned.so\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m20753774\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m3527736 (17.0%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m2.24 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_NHEM_77_GCCAAT_L007_R1_001.fastqAligned.so\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m26886833\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m5714425 (21.3%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m2.93 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_NHEM_77_GCCAAT_L007_R2_001.fastqAligned.so\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m30028063\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m5250224 (17.5%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m3.15 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_Sbcl2_64_trimmed_1.fastqAligned.sortedByCo\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m21244068\u001b[0m \u001b[0m                                            ||\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "||    Successfully assigned alignments : \u001b[36m3674801 (17.3%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m2.20 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_Sbcl2_64_trimmed_2.fastqAligned.sortedByCo\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m24486413\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m3264193 (13.3%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m2.33 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_Sbcl2_70_trimmed_1.fastqAligned.sortedByCo\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m19548518\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m3417976 (17.5%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m2.12 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_Sbcl2_70_trimmed_2.fastqAligned.sortedByCo\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m21870678\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m3079431 (14.1%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m2.20 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM1158_68_CGATGT_L008_1.fastqAligned.sorte\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m27824448\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m4590078 (16.5%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m2.87 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM1158_68_CGATGT_L008_2.fastqAligned.sorte\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m33119112\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m4186581 (12.6%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m3.10 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM1158_74_Trimmed_1.fastqAligned.sortedByC\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m13252130\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m2358344 (17.8%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m1.40 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM1158_74_Trimmed_2.fastqAligned.sortedByC\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m14855011\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m2125058 (14.3%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m1.43 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM1366_66_trimmed_1.fastqAligned.sortedByC\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m11732415\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m2047862 (17.5%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m1.20 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM1366_66_trimmed_2.fastqAligned.sortedByC\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m13425819\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m1832311 (13.6%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m1.26 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM1366_72_trimmed_1.fastqAligned.sortedByC\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m26248625\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m4574744 (17.4%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m2.63 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM1366_72_trimmed_2.fastqAligned.sortedByC\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m31839911\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m3983518 (12.5%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m2.94 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM3211_65_trimmed_1.fastqAligned.sortedByC\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m14042196\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m2511051 (17.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m1.42 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM3211_65_trimmed_2.fastqAligned.sortedByC\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m15935923\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m2295307 (14.4%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m1.48 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM3211_71_trimmed_1.fastqAligned.sortedByC\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m19709600\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m3441019 (17.5%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m1.95 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM3211_71_trimmed_2.fastqAligned.sortedByC\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m24368518\u001b[0m \u001b[0m                                            ||\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "||    Successfully assigned alignments : \u001b[36m2958097 (12.1%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m2.22 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM793_67_ATCACG_L008_R1_001.fastqAligned.s\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m16707017\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m3568967 (21.4%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m1.90 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM793_67_ATCACG_L008_R2_001.fastqAligned.s\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m19232245\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m3207645 (16.7%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m2.14 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM793_73_trimmed_1.fastqAligned.sortedByCo\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m17555428\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m3357102 (19.1%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m1.89 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM793_73_trimmed_2.fastqAligned.sortedByCo\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m19436092\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m3069827 (15.8%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m1.95 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM9_69_TTAGGC_L008_R1_001.fastqAligned.sor\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m30888112\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m6532574 (21.1%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m3.47 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM9_69_TTAGGC_L008_R2_001.fastqAligned.sor\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m36662632\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m5399439 (14.7%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m3.85 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM9_75_trimmed_1.fastqAligned.sortedByCoor\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m7240265\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m1328246 (18.3%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.76 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file mapped2RNA_WM9_75_trimmed_2.fastqAligned.sortedByCoor\u001b[0m ... \u001b[0m||\n",
      "||    WARNING: \u001b[36mSingle-end reads were found.\u001b[0m \u001b[0m                                  ||\n",
      "||    Total alignments : \u001b[36m7240265\u001b[0m \u001b[0m                                             ||\n",
      "||    Successfully assigned alignments : \u001b[36m1328246 (18.3%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.74 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"/home/stud03/NAS_MEB/co \u001b[0m ||\n",
      "|| untTable/featurecounts.txt.summary\" \u001b[0m                                       ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "featureCounts -p -T 23 -t exon -a ~/NAS_MEB/workspace/rnaseq/genome_ebigencode.v29.annotation.gtf -o ~/NAS_MEB/countTable/featurecounts.txt ~/NAS_MEB/mapped2/*.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "78bc4bae",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mfeaturecounts.txt\u001b[0m  \u001b[01;32mfeaturecounts.txt.summary\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "218aa4d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "featureCounts -T 23 -p -t exon -g gene_id -a /Users/mkappelmann-fenzl/GenomeIndices/Star/GRCh38_index_100/gencode.v24.primary_assembly.annotation.gtf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "ad30a772",
   "metadata": {},
   "outputs": [
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "grep 'gene_id \"\"' hg_ucsc.gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "49ae3ca1",
   "metadata": {},
   "outputs": [],
   "source": [
    "grep -v 'gene_id \"\"' hg_ucsc.gtf > hg_ucsc_fixed2.gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a477ba46",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd NAS_MEB/RNA_REFS_DIR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b3d9acf9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32mMaster_thesis.ipynb\u001b[0m  \u001b[01;32mUntitled.ipynb\u001b[0m  \u001b[01;34m_STARtmp\u001b[0m  \u001b[01;32mhg38.fa\u001b[0m  \u001b[01;32mhg_ucsc.gtf\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0c4ff891",
   "metadata": {},
   "outputs": [],
   "source": [
    "hg_ucsc.gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d77f314e",
   "metadata": {},
   "outputs": [],
   "source": [
    "grep 'gene_id \"\"' "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2b4334d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "07e078a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Command 'Python' not found, did you mean:\n",
      "\n",
      "  command 'jython' from deb jython (2.7.2~b2+repack1-1ubuntu1)\n",
      "  command 'cython' from deb cython (0.29.14-0.1ubuntu3)\n",
      "\n",
      "Try: apt install <deb name>\n",
      "\n"
     ]
    },
    {
     "ename": "",
     "evalue": "127",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "3beb2ae5",
   "metadata": {},
   "source": [
    "install R and added it to juypter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "638ed893",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: failed with initial frozen solve. Retrying with flexible solve.\n",
      "Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.\n",
      "Collecting package metadata (repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "\n",
      "==> WARNING: A newer version of conda exists. <==\n",
      "  current version: 4.11.0\n",
      "  latest version: 4.12.0\n",
      "\n",
      "Please update conda by running\n",
      "\n",
      "    $ conda update -n base conda\n",
      "\n",
      "\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/stud03/miniconda3\n",
      "\n",
      "  added / updated specs:\n",
      "    - r-irkernel\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    _r-mutex-1.0.0             |      anacondar_1           3 KB\n",
      "    binutils_impl_linux-64-2.35.1|       h27ae35d_9         3.6 MB\n",
      "    binutils_linux-64-2.35.1   |      h454624a_30          24 KB\n",
      "    bwidget-1.9.11             |                1         116 KB\n",
      "    ca-certificates-2022.2.1   |       h06a4308_0         122 KB\n",
      "    cairo-1.16.0               |       hf32fb01_1         1.0 MB\n",
      "    conda-4.12.0               |   py39h06a4308_0        14.5 MB\n",
      "    curl-7.69.1                |       hbc83047_0         137 KB\n",
      "    fontconfig-2.13.1          |       h6c09931_0         250 KB\n",
      "    fribidi-1.0.10             |       h7b6447c_0         103 KB\n",
      "    gcc_impl_linux-64-7.5.0    |      h7105cf2_17        41.9 MB\n",
      "    gcc_linux-64-7.5.0         |      h8f34230_30          25 KB\n",
      "    gfortran_impl_linux-64-7.5.0|      ha8c8e06_17         8.3 MB\n",
      "    gfortran_linux-64-7.5.0    |      h96bb648_30          24 KB\n",
      "    graphite2-1.3.14           |       h23475e2_0          99 KB\n",
      "    gsl-2.4                    |       h14c3975_4         1.8 MB\n",
      "    gxx_impl_linux-64-7.5.0    |      h0a5bf11_17         8.5 MB\n",
      "    gxx_linux-64-7.5.0         |      hffc177d_30          24 KB\n",
      "    harfbuzz-2.8.1             |       h6f93f22_0         1.1 MB\n",
      "    icu-58.2                   |       he6710b0_3        10.5 MB\n",
      "    jpeg-9d                    |       h7f8727e_0         232 KB\n",
      "    kernel-headers_linux-64-2.6.32|      he073ed8_15         707 KB  conda-forge\n",
      "    krb5-1.17.1                |       h173b8e3_0         1.3 MB\n",
      "    libcurl-7.69.1             |       h20c2e04_0         431 KB\n",
      "    libedit-3.1.20210714       |       h7f8727e_0         165 KB\n",
      "    libgcc-devel_linux-64-7.5.0|      hbbeae57_17         2.6 MB\n",
      "    libgfortran-ng-7.5.0       |      ha8ba4b0_17          22 KB\n",
      "    libgfortran4-7.5.0         |      ha8ba4b0_17         995 KB\n",
      "    libssh2-1.9.0              |       h1ba5d50_1         269 KB\n",
      "    libstdcxx-devel_linux-64-7.5.0|      hf0c5c8d_17         7.0 MB\n",
      "    libtiff-4.2.0              |       h85742a9_0         502 KB\n",
      "    libuuid-1.0.3              |       h7f8727e_2          17 KB\n",
      "    libwebp-base-1.2.2         |       h7f8727e_0         440 KB\n",
      "    libxml2-2.9.12             |       h03d6c58_0         1.2 MB\n",
      "    lz4-c-1.9.3                |       h295c915_1         185 KB\n",
      "    make-4.2.1                 |       h1bed415_1         415 KB\n",
      "    pango-1.45.3               |       hd140c19_0         361 KB\n",
      "    pixman-0.40.0              |       h7f8727e_1         373 KB\n",
      "    r-base-3.6.1               |       haffb61f_2        39.8 MB  r\n",
      "    r-base64enc-0.1_3          |    r36h96ca727_4          32 KB  r\n",
      "    r-crayon-1.3.4             |    r36h6115d3f_0         757 KB  r\n",
      "    r-digest-0.6.18            |    r36h96ca727_0         155 KB  r\n",
      "    r-evaluate-0.13            |    r36h6115d3f_0          83 KB  r\n",
      "    r-htmltools-0.3.6          |    r36h29659fb_0         184 KB  r\n",
      "    r-irdisplay-0.7.0          |    r36h6115d3f_0          38 KB  r\n",
      "    r-irkernel-0.8.15          |            r36_0         148 KB  r\n",
      "    r-jsonlite-1.6             |    r36h96ca727_0         1.0 MB  r\n",
      "    r-pbdzmq-0.3_3             |    r36h5455479_1         397 KB  r\n",
      "    r-rcpp-1.0.1               |    r36h29659fb_0         3.3 MB  r\n",
      "    r-repr-0.19.2              |    r36h6115d3f_0         143 KB  r\n",
      "    r-uuid-0.1_2               |    r36h96ca727_4          21 KB  r\n",
      "    sysroot_linux-64-2.12      |      he073ed8_15        31.4 MB  conda-forge\n",
      "    tktable-2.10               |       h14c3975_0          86 KB\n",
      "    zstd-1.4.9                 |       haebb681_0         480 KB\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:       187.1 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  _r-mutex           pkgs/r/linux-64::_r-mutex-1.0.0-anacondar_1\n",
      "  binutils_impl_lin~ pkgs/main/linux-64::binutils_impl_linux-64-2.35.1-h27ae35d_9\n",
      "  binutils_linux-64  pkgs/main/linux-64::binutils_linux-64-2.35.1-h454624a_30\n",
      "  bwidget            pkgs/main/linux-64::bwidget-1.9.11-1\n",
      "  cairo              pkgs/main/linux-64::cairo-1.16.0-hf32fb01_1\n",
      "  curl               pkgs/main/linux-64::curl-7.69.1-hbc83047_0\n",
      "  fontconfig         pkgs/main/linux-64::fontconfig-2.13.1-h6c09931_0\n",
      "  fribidi            pkgs/main/linux-64::fribidi-1.0.10-h7b6447c_0\n",
      "  gcc_impl_linux-64  pkgs/main/linux-64::gcc_impl_linux-64-7.5.0-h7105cf2_17\n",
      "  gcc_linux-64       pkgs/main/linux-64::gcc_linux-64-7.5.0-h8f34230_30\n",
      "  gfortran_impl_lin~ pkgs/main/linux-64::gfortran_impl_linux-64-7.5.0-ha8c8e06_17\n",
      "  gfortran_linux-64  pkgs/main/linux-64::gfortran_linux-64-7.5.0-h96bb648_30\n",
      "  graphite2          pkgs/main/linux-64::graphite2-1.3.14-h23475e2_0\n",
      "  gsl                pkgs/main/linux-64::gsl-2.4-h14c3975_4\n",
      "  gxx_impl_linux-64  pkgs/main/linux-64::gxx_impl_linux-64-7.5.0-h0a5bf11_17\n",
      "  gxx_linux-64       pkgs/main/linux-64::gxx_linux-64-7.5.0-hffc177d_30\n",
      "  harfbuzz           pkgs/main/linux-64::harfbuzz-2.8.1-h6f93f22_0\n",
      "  icu                pkgs/main/linux-64::icu-58.2-he6710b0_3\n",
      "  jpeg               pkgs/main/linux-64::jpeg-9d-h7f8727e_0\n",
      "  kernel-headers_li~ conda-forge/noarch::kernel-headers_linux-64-2.6.32-he073ed8_15\n",
      "  krb5               pkgs/main/linux-64::krb5-1.17.1-h173b8e3_0\n",
      "  libcurl            pkgs/main/linux-64::libcurl-7.69.1-h20c2e04_0\n",
      "  libedit            pkgs/main/linux-64::libedit-3.1.20210714-h7f8727e_0\n",
      "  libgcc-devel_linu~ pkgs/main/linux-64::libgcc-devel_linux-64-7.5.0-hbbeae57_17\n",
      "  libgfortran-ng     pkgs/main/linux-64::libgfortran-ng-7.5.0-ha8ba4b0_17\n",
      "  libgfortran4       pkgs/main/linux-64::libgfortran4-7.5.0-ha8ba4b0_17\n",
      "  libssh2            pkgs/main/linux-64::libssh2-1.9.0-h1ba5d50_1\n",
      "  libstdcxx-devel_l~ pkgs/main/linux-64::libstdcxx-devel_linux-64-7.5.0-hf0c5c8d_17\n",
      "  libtiff            pkgs/main/linux-64::libtiff-4.2.0-h85742a9_0\n",
      "  libuuid            pkgs/main/linux-64::libuuid-1.0.3-h7f8727e_2\n",
      "  libwebp-base       pkgs/main/linux-64::libwebp-base-1.2.2-h7f8727e_0\n",
      "  libxml2            pkgs/main/linux-64::libxml2-2.9.12-h03d6c58_0\n",
      "  lz4-c              pkgs/main/linux-64::lz4-c-1.9.3-h295c915_1\n",
      "  make               pkgs/main/linux-64::make-4.2.1-h1bed415_1\n",
      "  pango              pkgs/main/linux-64::pango-1.45.3-hd140c19_0\n",
      "  pixman             pkgs/main/linux-64::pixman-0.40.0-h7f8727e_1\n",
      "  r-base             r/linux-64::r-base-3.6.1-haffb61f_2\n",
      "  r-base64enc        r/linux-64::r-base64enc-0.1_3-r36h96ca727_4\n",
      "  r-crayon           r/noarch::r-crayon-1.3.4-r36h6115d3f_0\n",
      "  r-digest           r/linux-64::r-digest-0.6.18-r36h96ca727_0\n",
      "  r-evaluate         r/noarch::r-evaluate-0.13-r36h6115d3f_0\n",
      "  r-htmltools        r/linux-64::r-htmltools-0.3.6-r36h29659fb_0\n",
      "  r-irdisplay        r/noarch::r-irdisplay-0.7.0-r36h6115d3f_0\n",
      "  r-irkernel         r/noarch::r-irkernel-0.8.15-r36_0\n",
      "  r-jsonlite         r/linux-64::r-jsonlite-1.6-r36h96ca727_0\n",
      "  r-pbdzmq           r/linux-64::r-pbdzmq-0.3_3-r36h5455479_1\n",
      "  r-rcpp             r/linux-64::r-rcpp-1.0.1-r36h29659fb_0\n",
      "  r-repr             r/noarch::r-repr-0.19.2-r36h6115d3f_0\n",
      "  r-uuid             r/linux-64::r-uuid-0.1_2-r36h96ca727_4\n",
      "  sysroot_linux-64   conda-forge/noarch::sysroot_linux-64-2.12-he073ed8_15\n",
      "  tktable            pkgs/main/linux-64::tktable-2.10-h14c3975_0\n",
      "  zstd               pkgs/main/linux-64::zstd-1.4.9-haebb681_0\n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "  ca-certificates    conda-forge::ca-certificates-2021.10.~ --> pkgs/main::ca-certificates-2022.2.1-h06a4308_0\n",
      "  certifi            conda-forge::certifi-2021.10.8-py39hf~ --> pkgs/main::certifi-2021.10.8-py39h06a4308_2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  conda              conda-forge::conda-4.11.0-py39hf3d152~ --> pkgs/main::conda-4.12.0-py39h06a4308_0\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "conda install -c r r-irkernel\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7cae0acd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: failed with initial frozen solve. Retrying with flexible solve.\n",
      "Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.\n",
      "Collecting package metadata (repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "\n",
      "==> WARNING: A newer version of conda exists. <==\n",
      "  current version: 4.11.0\n",
      "  latest version: 4.12.0\n",
      "\n",
      "Please update conda by running\n",
      "\n",
      "    $ conda update -n base conda\n",
      "\n",
      "\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/stud03/miniconda3\n",
      "\n",
      "  added / updated specs:\n",
      "    - r\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    _r-mutex-1.0.0             |      anacondar_1           3 KB\n",
      "    binutils_impl_linux-64-2.35.1|       h27ae35d_9         3.6 MB\n",
      "    binutils_linux-64-2.35.1   |      h454624a_30          24 KB\n",
      "    bwidget-1.9.11             |                1         116 KB\n",
      "    ca-certificates-2022.2.1   |       h06a4308_0         122 KB\n",
      "    cairo-1.16.0               |       hf32fb01_1         1.0 MB\n",
      "    conda-4.12.0               |   py39h06a4308_0        14.5 MB\n",
      "    curl-7.69.1                |       hbc83047_0         137 KB\n",
      "    fontconfig-2.13.1          |       h6c09931_0         250 KB\n",
      "    fribidi-1.0.10             |       h7b6447c_0         103 KB\n",
      "    gcc_impl_linux-64-7.5.0    |      h7105cf2_17        41.9 MB\n",
      "    gcc_linux-64-7.5.0         |      h8f34230_30          25 KB\n",
      "    gfortran_impl_linux-64-7.5.0|      ha8c8e06_17         8.3 MB\n",
      "    gfortran_linux-64-7.5.0    |      h96bb648_30          24 KB\n",
      "    graphite2-1.3.14           |       h23475e2_0          99 KB\n",
      "    gsl-2.4                    |       h14c3975_4         1.8 MB\n",
      "    gxx_impl_linux-64-7.5.0    |      h0a5bf11_17         8.5 MB\n",
      "    gxx_linux-64-7.5.0         |      hffc177d_30          24 KB\n",
      "    harfbuzz-2.8.1             |       h6f93f22_0         1.1 MB\n",
      "    icu-58.2                   |       he6710b0_3        10.5 MB\n",
      "    jpeg-9d                    |       h7f8727e_0         232 KB\n",
      "    kernel-headers_linux-64-2.6.32|      he073ed8_15         707 KB  conda-forge\n",
      "    krb5-1.17.1                |       h173b8e3_0         1.3 MB\n",
      "    libcurl-7.69.1             |       h20c2e04_0         431 KB\n",
      "    libedit-3.1.20210714       |       h7f8727e_0         165 KB\n",
      "    libgcc-devel_linux-64-7.5.0|      hbbeae57_17         2.6 MB\n",
      "    libgfortran-ng-7.5.0       |      ha8ba4b0_17          22 KB\n",
      "    libgfortran4-7.5.0         |      ha8ba4b0_17         995 KB\n",
      "    libssh2-1.9.0              |       h1ba5d50_1         269 KB\n",
      "    libstdcxx-devel_linux-64-7.5.0|      hf0c5c8d_17         7.0 MB\n",
      "    libtiff-4.2.0              |       h85742a9_0         502 KB\n",
      "    libuuid-1.0.3              |       h7f8727e_2          17 KB\n",
      "    libwebp-base-1.2.2         |       h7f8727e_0         440 KB\n",
      "    libxml2-2.9.12             |       h03d6c58_0         1.2 MB\n",
      "    lz4-c-1.9.3                |       h295c915_1         185 KB\n",
      "    make-4.2.1                 |       h1bed415_1         415 KB\n",
      "    pango-1.45.3               |       hd140c19_0         361 KB\n",
      "    pixman-0.40.0              |       h7f8727e_1         373 KB\n",
      "    r-3.6.0                    |            r36_0           3 KB  r\n",
      "    r-base-3.6.1               |       haffb61f_2        39.8 MB  r\n",
      "    r-boot-1.3_20              |    r36h6115d3f_0         698 KB  r\n",
      "    r-class-7.3_15             |    r36h96ca727_0          93 KB  r\n",
      "    r-cluster-2.0.8            |    r36ha65eedd_0         550 KB  r\n",
      "    r-codetools-0.2_16         |    r36h6115d3f_0         107 KB  r\n",
      "    r-foreign-0.8_71           |    r36h96ca727_0         266 KB  r\n",
      "    r-kernsmooth-2.23_15       |    r36ha65eedd_4         102 KB  r\n",
      "    r-lattice-0.20_38          |    r36h96ca727_0         1.1 MB  r\n",
      "    r-mass-7.3_51.3            |    r36h96ca727_0         1.1 MB  r\n",
      "    r-matrix-1.2_17            |    r36h96ca727_0         3.8 MB  r\n",
      "    r-mgcv-1.8_28              |    r36h96ca727_0         2.6 MB  r\n",
      "    r-nlme-3.1_139             |    r36ha65eedd_0         2.2 MB  r\n",
      "    r-nnet-7.3_12              |    r36h96ca727_0         119 KB  r\n",
      "    r-recommended-3.6.0        |            r36_0           3 KB  r\n",
      "    r-rpart-4.1_15             |    r36h96ca727_0         719 KB  r\n",
      "    r-spatial-7.3_11           |    r36h96ca727_4         142 KB  r\n",
      "    r-survival-2.44_1.1        |    r36h96ca727_0         4.8 MB  r\n",
      "    sysroot_linux-64-2.12      |      he073ed8_15        31.4 MB  conda-forge\n",
      "    tktable-2.10               |       h14c3975_0          86 KB\n",
      "    zstd-1.4.9                 |       haebb681_0         480 KB\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:       199.2 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  _r-mutex           pkgs/r/linux-64::_r-mutex-1.0.0-anacondar_1\n",
      "  binutils_impl_lin~ pkgs/main/linux-64::binutils_impl_linux-64-2.35.1-h27ae35d_9\n",
      "  binutils_linux-64  pkgs/main/linux-64::binutils_linux-64-2.35.1-h454624a_30\n",
      "  bwidget            pkgs/main/linux-64::bwidget-1.9.11-1\n",
      "  cairo              pkgs/main/linux-64::cairo-1.16.0-hf32fb01_1\n",
      "  curl               pkgs/main/linux-64::curl-7.69.1-hbc83047_0\n",
      "  fontconfig         pkgs/main/linux-64::fontconfig-2.13.1-h6c09931_0\n",
      "  fribidi            pkgs/main/linux-64::fribidi-1.0.10-h7b6447c_0\n",
      "  gcc_impl_linux-64  pkgs/main/linux-64::gcc_impl_linux-64-7.5.0-h7105cf2_17\n",
      "  gcc_linux-64       pkgs/main/linux-64::gcc_linux-64-7.5.0-h8f34230_30\n",
      "  gfortran_impl_lin~ pkgs/main/linux-64::gfortran_impl_linux-64-7.5.0-ha8c8e06_17\n",
      "  gfortran_linux-64  pkgs/main/linux-64::gfortran_linux-64-7.5.0-h96bb648_30\n",
      "  graphite2          pkgs/main/linux-64::graphite2-1.3.14-h23475e2_0\n",
      "  gsl                pkgs/main/linux-64::gsl-2.4-h14c3975_4\n",
      "  gxx_impl_linux-64  pkgs/main/linux-64::gxx_impl_linux-64-7.5.0-h0a5bf11_17\n",
      "  gxx_linux-64       pkgs/main/linux-64::gxx_linux-64-7.5.0-hffc177d_30\n",
      "  harfbuzz           pkgs/main/linux-64::harfbuzz-2.8.1-h6f93f22_0\n",
      "  icu                pkgs/main/linux-64::icu-58.2-he6710b0_3\n",
      "  jpeg               pkgs/main/linux-64::jpeg-9d-h7f8727e_0\n",
      "  kernel-headers_li~ conda-forge/noarch::kernel-headers_linux-64-2.6.32-he073ed8_15\n",
      "  krb5               pkgs/main/linux-64::krb5-1.17.1-h173b8e3_0\n",
      "  libcurl            pkgs/main/linux-64::libcurl-7.69.1-h20c2e04_0\n",
      "  libedit            pkgs/main/linux-64::libedit-3.1.20210714-h7f8727e_0\n",
      "  libgcc-devel_linu~ pkgs/main/linux-64::libgcc-devel_linux-64-7.5.0-hbbeae57_17\n",
      "  libgfortran-ng     pkgs/main/linux-64::libgfortran-ng-7.5.0-ha8ba4b0_17\n",
      "  libgfortran4       pkgs/main/linux-64::libgfortran4-7.5.0-ha8ba4b0_17\n",
      "  libssh2            pkgs/main/linux-64::libssh2-1.9.0-h1ba5d50_1\n",
      "  libstdcxx-devel_l~ pkgs/main/linux-64::libstdcxx-devel_linux-64-7.5.0-hf0c5c8d_17\n",
      "  libtiff            pkgs/main/linux-64::libtiff-4.2.0-h85742a9_0\n",
      "  libuuid            pkgs/main/linux-64::libuuid-1.0.3-h7f8727e_2\n",
      "  libwebp-base       pkgs/main/linux-64::libwebp-base-1.2.2-h7f8727e_0\n",
      "  libxml2            pkgs/main/linux-64::libxml2-2.9.12-h03d6c58_0\n",
      "  lz4-c              pkgs/main/linux-64::lz4-c-1.9.3-h295c915_1\n",
      "  make               pkgs/main/linux-64::make-4.2.1-h1bed415_1\n",
      "  pango              pkgs/main/linux-64::pango-1.45.3-hd140c19_0\n",
      "  pixman             pkgs/main/linux-64::pixman-0.40.0-h7f8727e_1\n",
      "  r                  r/linux-64::r-3.6.0-r36_0\n",
      "  r-base             r/linux-64::r-base-3.6.1-haffb61f_2\n",
      "  r-boot             r/noarch::r-boot-1.3_20-r36h6115d3f_0\n",
      "  r-class            r/linux-64::r-class-7.3_15-r36h96ca727_0\n",
      "  r-cluster          r/linux-64::r-cluster-2.0.8-r36ha65eedd_0\n",
      "  r-codetools        r/noarch::r-codetools-0.2_16-r36h6115d3f_0\n",
      "  r-foreign          r/linux-64::r-foreign-0.8_71-r36h96ca727_0\n",
      "  r-kernsmooth       r/linux-64::r-kernsmooth-2.23_15-r36ha65eedd_4\n",
      "  r-lattice          r/linux-64::r-lattice-0.20_38-r36h96ca727_0\n",
      "  r-mass             r/linux-64::r-mass-7.3_51.3-r36h96ca727_0\n",
      "  r-matrix           r/linux-64::r-matrix-1.2_17-r36h96ca727_0\n",
      "  r-mgcv             r/linux-64::r-mgcv-1.8_28-r36h96ca727_0\n",
      "  r-nlme             r/linux-64::r-nlme-3.1_139-r36ha65eedd_0\n",
      "  r-nnet             r/linux-64::r-nnet-7.3_12-r36h96ca727_0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  r-recommended      r/linux-64::r-recommended-3.6.0-r36_0\n",
      "  r-rpart            r/linux-64::r-rpart-4.1_15-r36h96ca727_0\n",
      "  r-spatial          r/linux-64::r-spatial-7.3_11-r36h96ca727_4\n",
      "  r-survival         r/linux-64::r-survival-2.44_1.1-r36h96ca727_0\n",
      "  sysroot_linux-64   conda-forge/noarch::sysroot_linux-64-2.12-he073ed8_15\n",
      "  tktable            pkgs/main/linux-64::tktable-2.10-h14c3975_0\n",
      "  zstd               pkgs/main/linux-64::zstd-1.4.9-haebb681_0\n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "  ca-certificates    conda-forge::ca-certificates-2021.10.~ --> pkgs/main::ca-certificates-2022.2.1-h06a4308_0\n",
      "  certifi            conda-forge::certifi-2021.10.8-py39hf~ --> pkgs/main::certifi-2021.10.8-py39h06a4308_2\n",
      "  conda              conda-forge::conda-4.11.0-py39hf3d152~ --> pkgs/main::conda-4.12.0-py39h06a4308_0\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "conda install -c r r\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
